---
title: "TransitionTransversion2"
author: "Scott Leighow"
date: "June 26, 2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
rm(list=ls())
setwd("C:/Users/Scott/Box Sync/MutBias/TransitionTransversion2/")
library(ggplot2)
library(plotly)
```

## Import/QC data

```{r}

# AR in PCa
AR.df = read.csv("ARSummary2.csv",stringsAsFactors=F)
AR.df$gene = "AR"
AR.df = AR.df[,c(ncol(AR.df),1:(ncol(AR.df)-1))]

## Remove all multi-step mutations
AR.df = AR.df[AR.df$mut.codon!="multsub",]
AR.df$rel.freq = AR.df$freq/sum(AR.df$freq)

# Filter for identifiable transitions/transversions
AR.TsTv = AR.df[AR.df$class!="?",]

# Filter for identifiable substitutions
AR.mut = AR.df[AR.df$mut.codon!="?",]
AR.mut$tri.sub = paste(AR.mut$init.3nuc,AR.mut$pyr.sub,sep=".")
AR.mut$init.pyr = substr(AR.mut$pyr.sub,1,1)

# ESR1 in BCa
ESR1.df = read.csv("ESR1Summary.csv",stringsAsFactors=F)
ESR1.df$gene = "ESR1"
ESR1.df = ESR1.df[,c(ncol(ESR1.df),1:(ncol(ESR1.df)-1))]
ESR1.df = ESR1.df[ESR1.df$mut.codon!="multsub",]
ESR1.df$rel.freq = ESR1.df$freq/sum(ESR1.df$freq)

ESR1.TsTv = ESR1.df[ESR1.df$class!="?",]

ESR1.mut = ESR1.df[ESR1.df$mut.codon!="?",]
ESR1.mut$tri.sub = paste(ESR1.mut$init.3nuc,ESR1.mut$pyr.sub,sep=".")
ESR1.mut$init.pyr = substr(ESR1.mut$pyr.sub,1,1)

# KIT in GIST
KIT.df = read.csv("GISTSummary3.csv",stringsAsFactors=F)
KIT.df$gene = "KIT"
KIT.df = KIT.df[,c(ncol(KIT.df),1:(ncol(KIT.df)-1))]
KIT.df = KIT.df[KIT.df$mut.codon!="multsub",]
KIT.df$rel.freq = KIT.df$freq/sum(KIT.df$freq)

KIT.TsTv = KIT.df[KIT.df$class!="?",]

KIT.mut = KIT.df[KIT.df$mut.codon!="?",]
KIT.mut$tri.sub = paste(KIT.mut$init.3nuc,KIT.mut$pyr.sub,sep=".")
KIT.mut$init.pyr = substr(KIT.mut$pyr.sub,1,1)

# Combine data
tot.df = rbind(AR.df,ESR1.df,KIT.df)
tot.df$rel.freq = tot.df$freq/sum(tot.df$freq)

tot.TsTv = rbind(AR.TsTv,ESR1.TsTv,KIT.TsTv)
tot.TsTv$rel.freq = tot.TsTv$freq/sum(tot.TsTv$freq)

tot.mut = rbind(AR.mut,ESR1.mut,KIT.mut)
tot.mut$rel.freq = tot.mut$freq/sum(tot.mut$freq)

# # Remove all multi-step mutations
# tot.df = tot.df[tot.df$mut.codon!="multsub",]
# tot.df$rel.freq = tot.df$freq/sum(tot.df$freq)
# 
# # Filter for identifiable transitions/transversions
# tot.TsTv = tot.df[tot.df$class!="?",]
# 
# # Filter for identifiable mutations
# tot.mut = tot.df[tot.df$mut.codon!="?",]
# tot.mut$tri.sub = paste(tot.mut$init.3nuc,tot.mut$pyr.sub,sep=".")
# tot.mut$init.pyr = substr(tot.mut$pyr.sub,1,1)

```

## Allele distributions

```{r}

# Transitions/transversions

tot.TsTv.r = tot.TsTv[order(tot.TsTv$freq,decreasing=T),]
tot.TsTv.r$x = 1:nrow(tot.TsTv.r)

ggplot(tot.TsTv.r,aes(x=x,y=rel.freq))+theme_bw()+
  geom_bar(aes(fill=class),stat="identity")+
  scale_x_continuous(labels=c())+
  scale_fill_discrete("Class",labels=c("Transition","Transversion"))+
  ggtitle("Resistance Allele Distribution:\nObserved")+
  xlab("Resistance Variants")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=18,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=16,face="bold"),
    legend.position=c(0.8,0.8),
    legend.background = element_rect(fill="gray85")
  )

# Pyrmidine substitutions

tot.mut.r = tot.mut[order(tot.mut$freq,decreasing=T),]
tot.mut.r$x = 1:nrow(tot.mut.r)

ggplot(tot.mut.r[tot.mut.r$rel.freq>0,],aes(x=x,y=rel.freq))+theme_bw()+
  geom_bar(aes(fill=pyr.sub),stat="identity")+
  scale_x_continuous(labels=c())+
  scale_fill_discrete("Pyrmidine\nSubstitutions",labels=c("C>A","C>G","C>T","T>A","T>C","T>G"))+
  ggtitle("Resistance Allele Distribution:\nKIT, AR, and ESR1\nObserved")+
  xlab("Resistance Variants")+ylab("Frequency")+
  theme(
    # plot.title=element_text(size=24,face="bold",hjust=0.5),
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=22,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=22,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=22,face="bold"),
    legend.position=c(0.6,0.6),
    legend.background = element_rect(fill=alpha("gray85",0.5))
  )+
  guides(
    fill=guide_legend(nrow=3)
  )

ggsave("TotResDist1.png",width=6.5,height=6)

# Sense strand substitutions

ggplot(tot.mut.r,aes(x=x,y=rel.freq))+theme_bw()+
  geom_bar(aes(fill=nuc.sub),stat="identity")+
  scale_x_continuous(labels=c())+
  scale_fill_discrete("Sense Substitutions")+
  ggtitle("Resistance Allele Distribution:\nObserved")+
  xlab("Resistance Variants")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=18,face="bold"),
    legend.title.align=0.5,
    legend.position=c(0.6,0.75),
    legend.text=element_text(size=16,face="bold"),
    legend.background = element_rect(fill="gray85")
  )+
  guides(fill=guide_legend(nrow=3))

# Trinucleotide context

ggplot(tot.mut.r,aes(x=x,y=rel.freq))+theme_bw()+
  geom_bar(aes(fill=pyr.sub,color=init.3nuc),stat="identity")+
  scale_x_continuous(labels=c())+
  scale_fill_discrete("Pyrmidine Substitution")+
  scale_color_discrete("Trinucleotide Context")+
  ggtitle("Resistance Allele Distribution:\nObserved")+
  xlab("Resistance Variants")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=12,face="bold"),
    legend.title.align=0.5,
    legend.position=c(0.6,0.6),
    legend.text=element_text(size=10,face="bold"),
    legend.background = element_rect(fill="gray85")
  )+
  guides(
    fill=guide_legend(nrow=2),
    color=guide_legend(nrow=5)
  )


```

## Resistance mutational signature

```{r}

# Transitions/transversions

Ts.freq = sum(tot.TsTv$rel.freq[tot.TsTv$class=="Ts"])
Tv.freq = sum(tot.TsTv$rel.freq[tot.TsTv$class=="Tv"])

TsTv.freq = data.frame(class=c("Ts","Tv"),rel.freq=c(Ts.freq,Tv.freq))
ggplot(TsTv.freq,aes(x=class,y=rel.freq))+theme_bw()+
  geom_bar(aes(fill=class),stat="identity")+
  ggtitle("Transition/Transversion Mutational Signature")+
  scale_x_discrete(labels=c("Transitions","Transversions"))+
  xlab("")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=18,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=16,face="bold"),
    legend.position=c(0.8,0.8),
    legend.background = element_rect(fill="gray85")
  )+
  guides(fill=F)

## Pyrmidine substitutions

pyr.tab = table(tot.mut$pyr.sub)
pyr.subs = names(pyr.tab)
pyr.freq = pyr.tab
for (i in 1:length(pyr.tab)) {
  pyr.freq[i] = sum(tot.mut$rel.freq[tot.mut$pyr.sub==names(pyr.tab)[i]])
}

pyr.freq.df = as.data.frame(pyr.freq)
colnames(pyr.freq.df) = c("pyr.sub","rel.freq")

ggplot(pyr.freq.df,aes(x=pyr.sub,y=rel.freq))+theme_bw()+
  geom_hline(yintercept=1/6,color="gray",linetype="dashed",size=1.5)+
  geom_bar(aes(fill=pyr.sub),stat="identity")+
  ggtitle("Pyrmidine Mutational Signature")+
  scale_x_discrete(labels=pyr.subs)+
  xlab("")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=18,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=16,face="bold"),
    legend.position=c(0.8,0.8),
    legend.background = element_rect(fill="gray85")
  )+
  guides(fill=F)

## Sense strand substitutions

sns.tab = table(tot.mut$nuc.sub)
sns.subs = names(sns.tab)
sns.freq = sns.tab
for (i in 1:length(sns.tab)) {
  sns.freq[i] = sum(tot.mut$rel.freq[tot.mut$nuc.sub==names(sns.tab)[i]])
}

sns.freq.df = as.data.frame(sns.freq)
colnames(sns.freq.df) = c("sns.sub","rel.freq")

ggplot(sns.freq.df,aes(x=sns.sub,y=rel.freq))+theme_bw()+
  geom_hline(yintercept=1/12,color="gray",linetype="dashed",size=1.5)+
  geom_bar(aes(fill=sns.sub),stat="identity")+
  ggtitle("Sense Strand Mutational Signature")+
  scale_x_discrete(labels=sns.subs)+
  xlab("")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=18,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=16,face="bold"),
    legend.position=c(0.8,0.8),
    legend.background = element_rect(fill="gray85")
  )+
  guides(fill=F)

## Trinucleotide

# Vector of all 96 subs
tri.nuc.df = expand.grid(pos3 = c("A","C","T","G"),
                       pos2 = c("C","T"),
                       pos1 = c("A","C","T","G"))
tri.nuc.df = tri.nuc.df[,3:1]
tri.nucs = paste(tri.nuc.df$pos1,tri.nuc.df$pos2,tri.nuc.df$pos3,sep="")

tri.sub.df = expand.grid(tri.nuc=tri.nucs,pyr.sub=pyr.subs)
tri.sub.df$tri.pyr = substr(tri.sub.df$tri.nuc,2,2)
tri.sub.df$pyr.pyr = substr(tri.sub.df$pyr.sub,1,1)
tri.sub.df = tri.sub.df[tri.sub.df$tri.pyr==tri.sub.df$pyr.pyr,]
tri.subs = paste(tri.sub.df$tri.nuc,tri.sub.df$pyr.sub,sep=".")

tri.tab = table(tot.mut$tri.sub)
# fill in zero counts
tri.tab = tri.tab[tri.subs]
names(tri.tab) = tri.subs
tri.tab[is.na(tri.tab)] = 0

tri.freq = tri.tab
for (i in 1:length(tri.tab)) {
  tri.freq[i] = sum(tot.mut$rel.freq[tot.mut$tri.sub==names(tri.tab)[i]])
}

tri.freq.df = as.data.frame(tri.freq)
colnames(tri.freq.df) = c("tri.sub","rel.freq")
tri.freq.df$init.3nuc = substr(tri.freq.df$tri.sub,1,3)
tri.freq.df$init.pyr = substr(tri.freq.df$tri.sub,2,2)
tri.freq.df$pyr.sub = substr(tri.freq.df$tri.sub,5,6)

ggplot()+theme_bw()+
  geom_hline(yintercept=1/96,color="gray",linetype="dashed",size=1.5)+
  geom_bar(data=tri.freq.df[tri.freq.df$init.pyr=="C",],aes(x=init.3nuc,y=rel.freq,fill=pyr.sub),stat="identity")+
  facet_grid(.~pyr.sub)+
  # scale_x_discrete(labels=c())+
  ggtitle("Trinucleotide Mutational Signature")+
  # scale_x_discrete(labels=tri.nucs)+
  xlab("")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=14),
    axis.text.x=element_text(angle=90),
    legend.title=element_text(size=18,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=16,face="bold"),
    # strip.background=element_blank(),
    strip.text=element_text(size=12,face="bold")
  )+
  guides(fill=F)

ggplot()+theme_bw()+
  geom_hline(yintercept=1/96,color="gray",linetype="dashed",size=1.5)+
  geom_bar(data=tri.freq.df[tri.freq.df$init.pyr=="T",],aes(x=init.3nuc,y=rel.freq,fill=pyr.sub),stat="identity")+
  facet_grid(.~pyr.sub)+
  # scale_x_discrete(labels=c())+
  ggtitle("Trinucleotide Mutational Signature")+
  # scale_x_discrete(labels=tri.nucs)+
  xlab("")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=14),
    axis.text.x=element_text(angle=90),
    legend.title=element_text(size=18,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=16,face="bold"),
    # strip.background=element_blank(),
    strip.text=element_text(size=12,face="bold")
  )+
  guides(fill=F)

```

### Does this mutational signature differ from expected distribution under null models?

# Develop alternative models from mutational signatures

```{r}

# mutational signatures for each cancer type
mut.sig.df = read.csv("sigProfiler_exome_SBS_signatures.csv",stringsAsFactors=F)
cncr.types = read.csv("TCGA_WES_sigProfiler_SBS_signatures_in_samples.csv",stringsAsFactors=F)

# Breast cancer
BCa.df = cncr.types[cncr.types$Cancer.Types=="Breast-cancer",]
BCa.conv = colMeans(BCa.df[4:ncol(BCa.df)])
BCa.conv = BCa.conv/sum(BCa.conv)
BCa.sig = as.vector(as.matrix(mut.sig.df[3:ncol(mut.sig.df)]) %*% BCa.conv)
names(BCa.sig) = paste(mut.sig.df$SubType,".",substr(mut.sig.df$Type,1,1),substr(mut.sig.df$Type,3,3),sep="")
BCa.sig = BCa.sig[tri.subs]

# Prostate cancer
PCa.df = cncr.types[cncr.types$Cancer.Types=="Prost-AdenoCa",]
PCa.conv = colMeans(PCa.df[4:ncol(PCa.df)])
PCa.conv = PCa.conv/sum(PCa.conv)
PCa.sig = as.vector(as.matrix(mut.sig.df[3:ncol(mut.sig.df)]) %*% PCa.conv)
names(PCa.sig) = paste(mut.sig.df$SubType,".",substr(mut.sig.df$Type,1,1),substr(mut.sig.df$Type,3,3),sep="")
PCa.sig = PCa.sig[tri.subs]

# Sarcoma - no GIST-specific
GIST.df = cncr.types[cncr.types$Cancer.Types=="Sarcoma",]
GIST.conv = colMeans(GIST.df[4:ncol(GIST.df)])
GIST.conv = GIST.conv/sum(GIST.conv)
GIST.sig = as.vector(as.matrix(mut.sig.df[3:ncol(mut.sig.df)]) %*% GIST.conv)
names(GIST.sig) = paste(mut.sig.df$SubType,".",substr(mut.sig.df$Type,1,1),substr(mut.sig.df$Type,3,3),sep="")
GIST.sig = GIST.sig[tri.subs]

```

# Nonsynonymous mutations in genes - control for mutations that cause mutations
# (Refer to MutationAnalyzer070119.R)

```{r}

AR.nsyn = read.csv("ARNsynMutAnalysis.csv",stringsAsFactors=F)
ESR1.nsyn = read.csv("ESR1NsynMutAnalysis.csv",stringsAsFactors=F)
KIT.nsyn = read.csv("KITNsynMutAnalysis.csv",stringsAsFactors=F)

# Restrict to areas of resistance mutations
AR.nsyn.lim = AR.nsyn[AR.nsyn$idx>=516&AR.nsyn$idx<=734,] # note that indexing on resistance mutations differs from original transcript sequence
ESR1.nsyn.lim = ESR1.nsyn[ESR1.nsyn$idx>=329&ESR1.nsyn$idx<=546,]
KIT.nsyn.lim = KIT.nsyn[KIT.nsyn$idx>=559&KIT.nsyn$idx<=829,]

AR.nsyn.pyr = c()
ESR1.nsyn.pyr = c()
KIT.nsyn.pyr = c()
for (i in 1:length(pyr.subs)) {
  AR.nsyn.pyr = c(AR.nsyn.pyr,mean(AR.nsyn.lim$pyr.sub==pyr.subs[i]))
  ESR1.nsyn.pyr = c(ESR1.nsyn.pyr,mean(ESR1.nsyn.lim$pyr.sub==pyr.subs[i]))
  KIT.nsyn.pyr = c(KIT.nsyn.pyr,mean(KIT.nsyn.lim$pyr.sub==pyr.subs[i]))
}

```

# Transitions/transversions (2)

## Transitions/transversions: null model - Ts:Tv ratio = 0.5

```{r}

# set.seed(5)
# nsims = 1e4
# TsTv.null = as.data.frame(matrix(nrow=nsims,ncol=2))
# colnames(TsTv.null) = c("Ts","Tv")
# 
# temp.df = tot.TsTv[,c("class","rel.freq")]
# 
# # reassign and tally
# for (i in 1:nsims) {
#   
#   temp.df$class = sample(c("Ts","Tv"),nrow(temp.df),replace=T,prob=c(1/3,2/3))
#   
#   TsTv.null$Ts[i] = sum(temp.df$rel.freq[temp.df$class=="Ts"])
#   TsTv.null$Tv[i] = sum(temp.df$rel.freq[temp.df$class!="Ts"])
#   
# }
# 
# p.val = mean(TsTv.null$Ts>=Ts.freq) # 0.016
# 
# ggplot() + theme_bw()+
#   geom_histogram(data=TsTv.null,aes(x=Ts),fill="#4286f4")+
#   geom_segment(aes(x=Ts.freq,xend=Ts.freq,y=500,yend=90),color="red",arrow=arrow(length=unit(0.5,"cm")),size=2)+
#   geom_text(aes(x=Ts.freq,y=600,label=paste("Observed: ",round(Ts.freq,2),"\np-value=",round(p.val,3),sep="")),size=6,fontface="bold",color="red")+
#   ggtitle("Transition Distribution: Uniform Null Model")+
#   xlab("Transition Frequency")+ylab("")+
#   theme(
#     plot.title=element_text(size=20,face="bold",hjust=0.5),
#     axis.title=element_text(size=18,face="bold"),
#     axis.text=element_text(size=16)
#   )

```


## Transitions/transversions: null model - limit to amino acid changes

```{r}
# 
# set.seed(5)
# nsims = 1e4
# TsTv.null = as.data.frame(matrix(nrow=nsims,ncol=2))
# colnames(TsTv.null) = c("Ts","Tv")
# 
# temp.df = tot.TsTv[,c("class","rel.freq")]
# 
# # reassign and tally
# for (i in 1:nsims) {
#   
#   temp.df$class = sample(c("Ts","Tv"),nrow(temp.df),replace=T,prob=c(116/392,276/392))
#   
#   TsTv.null$Ts[i] = sum(temp.df$rel.freq[temp.df$class=="Ts"])
#   TsTv.null$Tv[i] = sum(temp.df$rel.freq[temp.df$class!="Ts"])
#   
# }
# 
# p.val = mean(TsTv.null$Ts>=Ts.freq) # 0.006
# 
# ggplot() + theme_bw()+
#   geom_histogram(data=TsTv.null,aes(x=Ts),fill="#4286f4")+
#   geom_segment(aes(x=Ts.freq,xend=Ts.freq,y=500,yend=90),color="red",arrow=arrow(length=unit(0.5,"cm")),size=2)+
#   geom_text(aes(x=Ts.freq,y=600,label=paste("Observed: ",round(Ts.freq,2),"\np-value=",round(p.val,3),sep="")),size=6,fontface="bold",color="red")+
#   ggtitle("Transition Distribution: AA Sub Null Model")+
#   xlab("Transition Frequency")+ylab("")+
#   theme(
#     plot.title=element_text(size=20,face="bold",hjust=0.5),
#     axis.title=element_text(size=18,face="bold"),
#     axis.text=element_text(size=16)
#   )


```

## Transition/transversion: null model - gene specific

```{r}
# 
# ## AR
# 
# AR.Ts = sum(AR.TsTv$count[AR.TsTv$class=="Ts"])
# 
# set.seed(5)
# nsims = 1e4
# AR.TsTv.null = as.data.frame(matrix(nrow=nsims,ncol=2))
# colnames(AR.TsTv.null) = c("Ts","Tv")
# 
# temp.df = AR.TsTv[,c("class","count")]
# 
# # reassign and tally
# for (i in 1:nsims) {
#   
#   temp.df$class = sample(c("Ts","Tv"),nrow(temp.df),replace=T,prob=c(116/392,276/392))
#   
#   AR.TsTv.null$Ts[i] = sum(temp.df$count[temp.df$class=="Ts"])
#   AR.TsTv.null$Tv[i] = sum(temp.df$count[temp.df$class!="Ts"])
#   
# }
# 
# p.val = mean(AR.TsTv.null$Ts>=AR.Ts) # 
# 
# ggplot() + theme_bw()+
#   geom_histogram(data=AR.TsTv.null,aes(x=Ts),fill="#4286f4")+
#   geom_segment(aes(x=AR.Ts,xend=AR.Ts,y=500,yend=90),color="red",arrow=arrow(length=unit(0.5,"cm")),size=2)+
#   geom_text(aes(x=AR.Ts,y=600,label=paste("Observed: ",round(AR.Ts,2),"\np-value=",round(p.val,3),sep="")),size=6,fontface="bold",color="red")+
#   ggtitle("AR Transition Distribution: Uniform Null Model")+
#   xlab("Transition Frequency")+ylab("")+
#   theme(
#     plot.title=element_text(size=20,face="bold",hjust=0.5),
#     axis.title=element_text(size=18,face="bold"),
#     axis.text=element_text(size=16)
#   )
# 
# ## ESR1
# 
# ESR1.Ts = sum(ESR1.TsTv$count[ESR1.TsTv$class=="Ts"])
# 
# set.seed(5)
# nsims = 1e4
# ESR1.TsTv.null = as.data.frame(matrix(nrow=nsims,ncol=2))
# colnames(ESR1.TsTv.null) = c("Ts","Tv")
# 
# temp.df = ESR1.TsTv[,c("class","count")]
# 
# # reassign and tally
# for (i in 1:nsims) {
#   
#   temp.df$class = sample(c("Ts","Tv"),nrow(temp.df),replace=T,prob=c(116/392,276/392))
#   
#   ESR1.TsTv.null$Ts[i] = sum(temp.df$count[temp.df$class=="Ts"])
#   ESR1.TsTv.null$Tv[i] = sum(temp.df$count[temp.df$class!="Ts"])
#   
# }
# 
# p.val = mean(ESR1.TsTv.null$Ts>=ESR1.Ts) # 
# 
# ggplot() + theme_bw()+
#   geom_histogram(data=ESR1.TsTv.null,aes(x=Ts),fill="#4286f4")+
#   geom_segment(aes(x=ESR1.Ts,xend=ESR1.Ts,y=500,yend=90),color="red",arrow=arrow(length=unit(0.5,"cm")),size=2)+
#   geom_text(aes(x=ESR1.Ts,y=600,label=paste("Observed: ",round(ESR1.Ts,2),"\np-value=",round(p.val,3),sep="")),size=6,fontface="bold",color="red")+
#   ggtitle("ESR1 Transition Distribution: Uniform Null Model")+
#   xlab("Transition Frequency")+ylab("")+
#   theme(
#     plot.title=element_text(size=20,face="bold",hjust=0.5),
#     axis.title=element_text(size=18,face="bold"),
#     axis.text=element_text(size=16)
#   )
# 
# ## KIT
# 
# KIT.Ts = sum(KIT.TsTv$count[KIT.TsTv$class=="Ts"])
# 
# set.seed(5)
# nsims = 1e4
# KIT.TsTv.null = as.data.frame(matrix(nrow=nsims,ncol=2))
# colnames(KIT.TsTv.null) = c("Ts","Tv")
# 
# temp.df = KIT.TsTv[,c("class","count")]
# 
# # reassign and tally
# for (i in 1:nsims) {
#   
#   temp.df$class = sample(c("Ts","Tv"),nrow(temp.df),replace=T,prob=c(116/392,276/392))
#   
#   KIT.TsTv.null$Ts[i] = sum(temp.df$count[temp.df$class=="Ts"])
#   KIT.TsTv.null$Tv[i] = sum(temp.df$count[temp.df$class!="Ts"])
#   
# }
# 
# p.val = mean(KIT.TsTv.null$Ts>=KIT.Ts) # 
# 
# ggplot() + theme_bw()+
#   geom_histogram(data=KIT.TsTv.null,aes(x=Ts),fill="#4286f4")+
#   geom_segment(aes(x=KIT.Ts,xend=KIT.Ts,y=500,yend=90),color="red",arrow=arrow(length=unit(0.5,"cm")),size=2)+
#   geom_text(aes(x=KIT.Ts,y=600,label=paste("Observed: ",round(KIT.Ts,2),"\np-value=",round(p.val,3),sep="")),size=6,fontface="bold",color="red")+
#   ggtitle("KIT Transition Distribution: Uniform Null Model")+
#   xlab("Transition Frequency")+ylab("")+
#   theme(
#     plot.title=element_text(size=20,face="bold",hjust=0.5),
#     axis.title=element_text(size=18,face="bold"),
#     axis.text=element_text(size=16)
#   )

```

## Transitions/transversions: null model - conserve transition path frequency

```{r}

n.Ts.path = sum(tot.TsTv$class=="Ts")
n.Tv.path = sum(tot.TsTv$class=="Tv")

set.seed(5)
nsims = 1e4
TsTv.null = as.data.frame(matrix(nrow=nsims,ncol=2))
colnames(TsTv.null) = c("Ts","Tv")

temp.df = tot.TsTv[,c("class","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$class = sample(c("Ts","Tv"),nrow(temp.df),replace=T,prob=c(n.Ts.path,n.Tv.path))
  
  TsTv.null$Ts[i] = sum(temp.df$rel.freq[temp.df$class=="Ts"])
  TsTv.null$Tv[i] = sum(temp.df$rel.freq[temp.df$class!="Ts"])
  
}

p.val = mean(TsTv.null$Ts>=Ts.freq) # 0.006

ggplot() + theme_bw()+
  geom_histogram(data=TsTv.null,aes(x=Ts),fill="#4286f4")+
  geom_segment(aes(x=Ts.freq,xend=Ts.freq,y=500,yend=300),color="red",arrow=arrow(length=unit(0.5,"cm")),size=2)+
  geom_text(aes(x=Ts.freq,y=600,label=paste("Observed: ",round(Ts.freq,2),"\np-value=",round(p.val,3),sep="")),size=6,fontface="bold",color="red")+
  ggtitle("Transition Distribution: Conserved Path Model")+
  xlab("Transition Frequency")+ylab("")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )


```

## Transitions/transversions: null model - control for nonsynonymous mutations across genes

```{r}

# add .lim (AR.nsyn ==> AR.nsyn.lim) to consider only regions with mutations
AR.Ts = mean(AR.nsyn.lim$TsTv=="Ts")
ESR1.Ts = mean(ESR1.nsyn.lim$TsTv=="Ts")
KIT.Ts = mean(KIT.nsyn.lim$TsTv=="Ts")


set.seed(5)
nsims = 1e4
TsTv.null = as.data.frame(matrix(nrow=nsims,ncol=2))
colnames(TsTv.null) = c("Ts","Tv")

temp.df = tot.TsTv[,c("class","rel.freq","gene")]

# reassign and tally
for (i in 1:nsims) {
  
  # temp.df$class = sample(c("Ts","Tv"),nrow(temp.df),replace=T,prob=c(116/392,276/392))
  temp.df$class[temp.df$gene=="AR"] = sample(c("Ts","Tv"),sum(temp.df$gene=="AR"),replace=T,prob=c(AR.Ts,1-AR.Ts))
  temp.df$class[temp.df$gene=="ESR1"] = sample(c("Ts","Tv"),sum(temp.df$gene=="ESR1"),replace=T,prob=c(ESR1.Ts,1-ESR1.Ts))
  temp.df$class[temp.df$gene=="KIT"] = sample(c("Ts","Tv"),sum(temp.df$gene=="KIT"),replace=T,prob=c(KIT.Ts,1-KIT.Ts))

  
  TsTv.null$Ts[i] = sum(temp.df$rel.freq[temp.df$class=="Ts"])
  TsTv.null$Tv[i] = sum(temp.df$rel.freq[temp.df$class!="Ts"])
  
}

p.val = mean(TsTv.null$Ts>=Ts.freq) # 0.006

ggplot() + theme_bw()+
  geom_histogram(data=TsTv.null,aes(x=Ts),fill="#4286f4")+
  geom_segment(aes(x=Ts.freq,xend=Ts.freq,y=500,yend=90),color="red",arrow=arrow(length=unit(0.5,"cm")),size=2)+
  geom_text(aes(x=Ts.freq,y=600,label=paste("Observed: ",round(Ts.freq,2),"\np-value=",round(p.val,3),sep="")),size=6,fontface="bold",color="red")+
  ggtitle("Transition Distribution: Nonsyn Sub Null Model")+
  xlab("Transition Frequency")+ylab("")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )


```

## Transitions/transversions: alternative model

```{r}

BCa.Ts = sum(BCa.sig[substr(names(BCa.sig),5,6)%in%c("CT","TC")])
BCa.Tv = sum(BCa.sig[!substr(names(BCa.sig),5,6)%in%c("CT","TC")])

PCa.Ts = sum(PCa.sig[substr(names(PCa.sig),5,6)%in%c("CT","TC")])
PCa.Tv = sum(PCa.sig[!substr(names(PCa.sig),5,6)%in%c("CT","TC")])

GIST.Ts = sum(GIST.sig[substr(names(GIST.sig),5,6)%in%c("CT","TC")])
GIST.Tv = sum(GIST.sig[!substr(names(GIST.sig),5,6)%in%c("CT","TC")])

set.seed(5)
nsims = 1e4
TsTv.alt = as.data.frame(matrix(nrow=nsims,ncol=2))
colnames(TsTv.alt) = c("Ts","Tv")

temp.df = tot.TsTv[,c("class","rel.freq","gene")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$class[temp.df$gene=="AR"] = sample(c("Ts","Tv"),nrow(temp.df[temp.df$gene=="AR",]),replace=T,prob=c(PCa.Ts,PCa.Tv))
  temp.df$class[temp.df$gene=="ESR1"] = sample(c("Ts","Tv"),nrow(temp.df[temp.df$gene=="ESR1",]),replace=T,prob=c(BCa.Ts,BCa.Tv))
  temp.df$class[temp.df$gene=="KIT"] = sample(c("Ts","Tv"),nrow(temp.df[temp.df$gene=="KIT",]),replace=T,prob=c(GIST.Ts,GIST.Tv))

  
  
  TsTv.alt$Ts[i] = sum(temp.df$rel.freq[temp.df$class=="Ts"])
  TsTv.alt$Tv[i] = sum(temp.df$rel.freq[temp.df$class!="Ts"])
  
}

p.val = mean(TsTv.alt$Ts>=Ts.freq) # 0.66

ggplot() + theme_bw()+
  geom_histogram(data=TsTv.alt,aes(x=Ts),fill="#4286f4")+
  geom_segment(aes(x=Ts.freq,xend=Ts.freq,y=500,yend=90),color="red",arrow=arrow(length=unit(0.5,"cm")),size=2)+
  geom_text(aes(x=Ts.freq,y=600,label=paste("Observed: ",round(Ts.freq,2),"\np-value=",round(p.val,3),sep="")),size=6,fontface="bold",color="red")+
  ggtitle("Transition Distribution: Alternative Model")+
  xlab("Transition Frequency")+ylab("")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )


```

# Pyrmidine substitutions (6)

## Pyrmidine substitutions: null model - uniform distribution

```{r}

set.seed(5)
nsims = 1e4
pyr.sub.null = as.data.frame(matrix(nrow=nsims,ncol=6))
colnames(pyr.sub.null) = pyr.subs

temp.df = tot.mut[,c("pyr.sub","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$pyr.sub = sample(pyr.subs,nrow(temp.df),replace=T,prob=rep(1/6,6))
  
  for (j in 1:length(pyr.subs)) {
    
    pyr.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$pyr.sub==pyr.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

pyr.mat = as.matrix(pyr.sub.null[1:5])
D2 = mahalanobis(pyr.mat,colMeans(pyr.mat),cov(pyr.mat))

pyr.obs = as.vector(pyr.freq.df$rel.freq)
names(pyr.obs) = pyr.freq.df$pyr.sub
D2.obs = mahalanobis(pyr.obs[1:5],colMeans(pyr.mat),cov(pyr.mat))

p.val = mean(D2>=D2.obs) #0.059

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=450,yend=150),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=550,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("Pyrmidine Substitution: Uniform Null Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(pyr.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(pyr.obs[1:5])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```

## Pyrmidine substitutions: null model - control for gene GC content

```{r}

AR.GC.con = 0.57
ESR1.GC.con = 0.57
KIT.GC.con = 0.45

set.seed(5)
nsims = 1e4
pyr.sub.null = as.data.frame(matrix(nrow=nsims,ncol=6))
colnames(pyr.sub.null) = pyr.subs

temp.df = tot.mut[,c("pyr.sub","rel.freq","gene")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$pyr.sub[temp.df$gene=="AR"] = sample(pyr.subs,nrow(temp.df[temp.df$gene=="AR",]),replace=T,prob=c(rep(AR.GC.con,3),rep(1-AR.GC.con,3)))
  temp.df$pyr.sub[temp.df$gene=="ESR1"] = sample(pyr.subs,nrow(temp.df[temp.df$gene=="ESR1",]),replace=T,prob=c(rep(ESR1.GC.con,3),rep(1-ESR1.GC.con,3)))
  temp.df$pyr.sub[temp.df$gene=="KIT"] = sample(pyr.subs,nrow(temp.df[temp.df$gene=="KIT",]),replace=T,prob=c(rep(KIT.GC.con,3),rep(1-KIT.GC.con,3)))
  
  for (j in 1:length(pyr.subs)) {
    
    pyr.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$pyr.sub==pyr.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

pyr.mat = as.matrix(pyr.sub.null[1:5])
D2 = mahalanobis(pyr.mat,colMeans(pyr.mat),cov(pyr.mat))

pyr.obs = as.vector(pyr.freq.df$rel.freq)
names(pyr.obs) = pyr.freq.df$pyr.sub
D2.obs = mahalanobis(pyr.obs[1:5],colMeans(pyr.mat),cov(pyr.mat))

p.val = mean(D2>=D2.obs) #0.036
tot.pval = p.val

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=450,yend=150),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=550,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("Pyrmidine Substitutions: GC Null Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(pyr.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(pyr.obs[1:5])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```

## Pyrmidine substitutions: null model - control for gene GC content, gene specific

```{r}

## AR

AR.GC.con = 0.57

AR.pyr.tab = table(AR.mut$pyr.sub)
AR.pyr.subs = names(AR.pyr.tab)
AR.pyr.freq = AR.pyr.tab
for (i in 1:length(AR.pyr.tab)) {
  AR.pyr.freq[i] = sum(AR.mut$count[AR.mut$pyr.sub==names(pyr.tab)[i]])
}

AR.pyr.freq.df = as.data.frame(AR.pyr.freq)
colnames(AR.pyr.freq.df) = c("pyr.sub","count")

set.seed(5)
nsims = 1e4
pyr.sub.null = as.data.frame(matrix(nrow=nsims,ncol=6))
colnames(pyr.sub.null) = pyr.subs

temp.df = tot.mut[tot.df$gene=="AR",c("pyr.sub","count")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$pyr.sub = sample(pyr.subs,nrow(temp.df),replace=T,prob=c(rep(AR.GC.con,3),rep(1-AR.GC.con,3)))
  
  for (j in 1:length(pyr.subs)) {
    
    pyr.sub.null[i,j] = sum(temp.df$count[temp.df$pyr.sub==pyr.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

pyr.mat = as.matrix(pyr.sub.null[1:5])
D2 = mahalanobis(pyr.mat,colMeans(pyr.mat),cov(pyr.mat))

pyr.obs = as.vector(AR.pyr.freq.df$count)
names(pyr.obs) = AR.pyr.freq.df$pyr.sub
D2.obs = mahalanobis(pyr.obs[1:5],colMeans(pyr.mat),cov(pyr.mat))

p.val = mean(D2>=D2.obs) #0.036

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=450,yend=150),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=550,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("AR Pyrmidine Substitutions: GC Null Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(pyr.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(pyr.obs[1:5])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("AR Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )


## ESR1

ESR1.GC.con = 0.57

ESR1.pyr.tab = table(ESR1.mut$pyr.sub)
ESR1.pyr.subs = names(ESR1.pyr.tab)
ESR1.pyr.freq = ESR1.pyr.tab
for (i in 1:length(ESR1.pyr.tab)) {
  ESR1.pyr.freq[i] = sum(ESR1.mut$count[ESR1.mut$pyr.sub==names(pyr.tab)[i]])
}

ESR1.pyr.freq.df = as.data.frame(ESR1.pyr.freq)
colnames(ESR1.pyr.freq.df) = c("pyr.sub","count")

set.seed(5)
nsims = 1e4
pyr.sub.null = as.data.frame(matrix(nrow=nsims,ncol=6))
colnames(pyr.sub.null) = pyr.subs

temp.df = tot.mut[tot.df$gene=="ESR1",c("pyr.sub","count")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$pyr.sub = sample(pyr.subs,nrow(temp.df),replace=T,prob=c(rep(ESR1.GC.con,3),rep(1-ESR1.GC.con,3)))
  
  for (j in 1:length(pyr.subs)) {
    
    pyr.sub.null[i,j] = sum(temp.df$count[temp.df$pyr.sub==pyr.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

pyr.mat = as.matrix(pyr.sub.null[1:5])
D2 = mahalanobis(pyr.mat,colMeans(pyr.mat),cov(pyr.mat))

pyr.obs = as.vector(ESR1.pyr.freq.df$count)
names(pyr.obs) = ESR1.pyr.freq.df$pyr.sub
D2.obs = mahalanobis(pyr.obs[1:5],colMeans(pyr.mat),cov(pyr.mat))

p.val = mean(D2>=D2.obs) #0.036

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=450,yend=150),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=550,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("ESR1 Pyrmidine Substitutions: GC Null Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(pyr.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(pyr.obs[1:5])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("ESR1 Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

## GIST

KIT.pyr.tab = table(KIT.mut$pyr.sub)
KIT.pyr.subs = names(KIT.pyr.tab)
KIT.pyr.freq = KIT.pyr.tab
for (i in 1:length(KIT.pyr.tab)) {
  KIT.pyr.freq[i] = sum(KIT.mut$count[KIT.mut$pyr.sub==names(pyr.tab)[i]])
}


```


## Transitions/transversions: null model - control for nonsynonymous mutations across genes

```{r}

AR.prob = c()
ESR1.prob = c()
KIT.prob = c()

for (i in 1:length(pyr.subs)) {
  AR.prob = c(AR.prob,mean(AR.nsyn.lim$pyr.sub==pyr.subs[i]))
  ESR1.prob = c(ESR1.prob,mean(ESR1.nsyn.lim$pyr.sub==pyr.subs[i]))
  KIT.prob = c(KIT.prob,mean(KIT.nsyn.lim$pyr.sub==pyr.subs[i]))

}

set.seed(5)
nsims = 1e4
pyr.sub.null = as.data.frame(matrix(nrow=nsims,ncol=6))
colnames(pyr.sub.null) = pyr.subs

temp.df = tot.mut[,c("pyr.sub","rel.freq","gene")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$pyr.sub[temp.df$gene=="AR"] = sample(pyr.subs,nrow(temp.df[temp.df$gene=="AR",]),replace=T,prob=AR.prob)
  temp.df$pyr.sub[temp.df$gene=="ESR1"] = sample(pyr.subs,nrow(temp.df[temp.df$gene=="ESR1",]),replace=T,prob=ESR1.prob)
  temp.df$pyr.sub[temp.df$gene=="KIT"] = sample(pyr.subs,nrow(temp.df[temp.df$gene=="KIT",]),replace=T,prob=KIT.prob)
  
  for (j in 1:length(pyr.subs)) {
    
    pyr.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$pyr.sub==pyr.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

pyr.mat = as.matrix(pyr.sub.null[1:5])
D2 = mahalanobis(pyr.mat,colMeans(pyr.mat),cov(pyr.mat))

pyr.obs = as.vector(pyr.freq.df$rel.freq)
names(pyr.obs) = pyr.freq.df$pyr.sub
D2.obs = mahalanobis(pyr.obs[1:5],colMeans(pyr.mat),cov(pyr.mat))

p.val = mean(D2>=D2.obs) #0.036

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=450,yend=150),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=550,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("Pyrmidine Substitutions: Nonsyn Null Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(pyr.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(pyr.obs[1:5])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```

## Pyrmidine substitutions: null model - control for amino acid substitutions

```{r}

set.seed(5)
nsims = 1e4
pyr.sub.null = as.data.frame(matrix(nrow=nsims,ncol=6))
colnames(pyr.sub.null) = pyr.subs

temp.df = tot.mut[,c("pyr.sub","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$pyr.sub = sample(pyr.subs,nrow(temp.df),replace=T,prob=c(68,76,58,64,58,68))
  
  for (j in 1:length(pyr.subs)) {
    
    pyr.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$pyr.sub==pyr.subs[j]])
    
  }
  
}

temp.df.r = temp.df[order(temp.df$rel.freq,decreasing=T),]
temp.df.r$x = 1:nrow(temp.df.r)
ggplot(temp.df.r[temp.df.r$rel.freq>0,],aes(x=x,y=rel.freq))+theme_bw()+
  geom_bar(aes(fill=pyr.sub),stat="identity")+
  scale_x_continuous(labels=c())+
  scale_fill_discrete("Pyrmidine\nSubstitutions",labels=c("C>A","C>G","C>T","T>A","T>C","T>G"))+
  ggtitle("Resistance Allele Distribution:\nKIT, AR, and ESR1\nNull Model Simulation Example")+
  xlab("Resistance Variants")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=22,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=22,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=22,face="bold"),
    legend.position=c(0.6,0.6),
    legend.background = element_rect(fill=alpha("gray85",0.5))
  )+
  guides(
    fill=guide_legend(nrow=3)
  )

ggsave("TotResNullDist.png",width=6.5,height=6)

# calculate mahalanobis dist

pyr.mat = as.matrix(pyr.sub.null[1:5])
D2 = mahalanobis(pyr.mat,colMeans(pyr.mat),cov(pyr.mat))

pyr.obs = as.vector(pyr.freq.df$rel.freq)
names(pyr.obs) = pyr.freq.df$pyr.sub
D2.obs = mahalanobis(pyr.obs[1:5],colMeans(pyr.mat),cov(pyr.mat))

p.val = mean(D2>=D2.obs) #0.059

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=450,yend=70),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=550,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=8,fontface="bold",color="red")+
  ggtitle("Null Model Pyrmidine Substitution:\nKIT, AR, and ESR1")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=22,face="bold"),
    axis.text=element_text(size=16)
  )

ggsave("TotMahalnobisHist.png",width=6.5,height=6)

pca = prcomp(rbind(pyr.mat,pyr.obs[1:5]),scale=F)
pca.obs = predict(pca,newdata = t(data.frame(pyr.obs[1:5])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC3),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

ggplot(pyr.sub.null)+theme_bw()+
  geom_point(aes(TC,CT),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pyr.obs["TC"],pyr.obs["CT"]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

# sim.pca = pca.df
# sim.pca$res = "sim"
# obs.pca = as.data.frame(pca.obs)
# obs.pca$res = "obs"
# pca.cmplt = rbind(sim.pca,obs.pca)

write.table(pca.df,"totPCA.csv",row.names=F,col.names=F,sep=",")
write.table(pca.obs,"totPCAobs.csv",row.names=F,col.names=F,sep=",")

# plot_ly(pca.cmplt, x = ~PC1, y = ~PC2, z = ~PC3,color=~res, colors = c('red','gray'),opacity=0.4) %>%
#   add_markers() %>%
#   layout(scene = list(xaxis = list(title = 'PC1'),
#                      yaxis = list(title = 'PC2'),
#                      zaxis = list(title = 'PC3')))

```

## Pyrmidine substitutions: alternative model

```{r}

# pyrmidine substitution biases from signatures
BCa.pyr.sig = rep(NA,6)
names(BCa.pyr.sig) = pyr.subs
for (i in 1:length(pyr.subs)) {
  BCa.pyr.sig[i] = sum(BCa.sig[substr(names(BCa.sig),5,6)==pyr.subs[i]])
}

PCa.pyr.sig = (matrix(nrow=1,ncol=6))
colnames(PCa.pyr.sig) = pyr.subs
for (i in 1:length(pyr.subs)) {
  PCa.pyr.sig[,i] = sum(PCa.sig[substr(names(PCa.sig),5,6)==pyr.subs[i]])
}

GIST.pyr.sig = (matrix(nrow=1,ncol=6))
colnames(GIST.pyr.sig) = pyr.subs
for (i in 1:length(pyr.subs)) {
  GIST.pyr.sig[,i] = sum(GIST.sig[substr(names(GIST.sig),5,6)==pyr.subs[i]])
}

# correct for GC content
ESR1.pyr.sig = c(BCa.pyr.sig[1:3]*(ESR1.GC.con),BCa.pyr.sig[4:6]*(1-ESR1.GC.con))
ESR1.pyr.sig = ESR1.pyr.sig/sum(ESR1.pyr.sig)

AR.pyr.sig = c(PCa.pyr.sig[1:3]*(AR.GC.con),PCa.pyr.sig[4:6]*(1-AR.GC.con))
AR.pyr.sig = AR.pyr.sig/sum(AR.pyr.sig)

KIT.pyr.sig = c(GIST.pyr.sig[1:3]*(KIT.GC.con),GIST.pyr.sig[4:6]*(1-KIT.GC.con))
KIT.pyr.sig = KIT.pyr.sig/sum(KIT.pyr.sig)

set.seed(5)
nsims = 1e4
pyr.sub.alt = as.data.frame(matrix(nrow=nsims,ncol=6))
colnames(pyr.sub.alt) = pyr.subs

temp.df = tot.mut[,c("pyr.sub","rel.freq","gene")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$pyr.sub[temp.df$gene=="AR"] = sample(pyr.subs,nrow(temp.df[temp.df$gene=="AR",]),replace=T,prob=AR.pyr.sig)
  temp.df$pyr.sub[temp.df$gene=="ESR1"] = sample(pyr.subs,nrow(temp.df[temp.df$gene=="ESR1",]),replace=T,prob=ESR1.pyr.sig)
  temp.df$pyr.sub[temp.df$gene=="KIT"] = sample(pyr.subs,nrow(temp.df[temp.df$gene=="KIT",]),replace=T,prob=KIT.pyr.sig)
  
  for (j in 1:length(pyr.subs)) {
    
    pyr.sub.alt[i,j] = sum(temp.df$rel.freq[temp.df$pyr.sub==pyr.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

pyr.mat = as.matrix(pyr.sub.alt[1:5])
D2 = mahalanobis(pyr.mat,colMeans(pyr.mat),cov(pyr.mat))

pyr.obs = as.vector(pyr.freq.df$rel.freq)
names(pyr.obs) = pyr.freq.df$pyr.sub
D2.obs = mahalanobis(pyr.obs[1:5],colMeans(pyr.mat),cov(pyr.mat))

p.val = mean(D2>=D2.obs) #0.055

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=450,yend=150),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=550,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("Pyrmidine Substitution: Alternative Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(pyr.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(pyr.obs[1:5])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```

## Pyrmidine substitutions: alternative model, consider nonsynonymous mutations

```{r}

AR.nsyn.tab = table(AR.nsyn.lim$pyr.sub)
AR.nsyn.tab = AR.nsyn.tab[pyr.subs]
AR.nsyn.pyr.sig = PCa.pyr.sig*as.vector(AR.nsyn.tab)
AR.nsyn.pyr.sig = AR.nsyn.pyr.sig/sum(AR.nsyn.pyr.sig)

ESR1.nsyn.tab = table(ESR1.nsyn.lim$pyr.sub)
ESR1.nsyn.tab = ESR1.nsyn.tab[pyr.subs]
ESR1.nsyn.pyr.sig = BCa.pyr.sig*as.vector(ESR1.nsyn.tab)
ESR1.nsyn.pyr.sig = ESR1.nsyn.pyr.sig/sum(ESR1.nsyn.pyr.sig)

KIT.nsyn.tab = table(KIT.nsyn.lim$pyr.sub)
KIT.nsyn.tab = KIT.nsyn.tab[pyr.subs]
KIT.nsyn.pyr.sig = GIST.pyr.sig*as.vector(KIT.nsyn.tab)
KIT.nsyn.pyr.sig = KIT.nsyn.pyr.sig/sum(KIT.nsyn.pyr.sig)


set.seed(5)
nsims = 1e4
pyr.sub.alt = as.data.frame(matrix(nrow=nsims,ncol=6))
colnames(pyr.sub.alt) = pyr.subs

temp.df = tot.mut[,c("pyr.sub","rel.freq","gene")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$pyr.sub[temp.df$gene=="AR"] = sample(pyr.subs,nrow(temp.df[temp.df$gene=="AR",]),replace=T,prob=AR.nsyn.pyr.sig)
  temp.df$pyr.sub[temp.df$gene=="ESR1"] = sample(pyr.subs,nrow(temp.df[temp.df$gene=="ESR1",]),replace=T,prob=ESR1.nsyn.pyr.sig)
  temp.df$pyr.sub[temp.df$gene=="KIT"] = sample(pyr.subs,nrow(temp.df[temp.df$gene=="KIT",]),replace=T,prob=KIT.nsyn.pyr.sig)
  
  for (j in 1:length(pyr.subs)) {
    
    pyr.sub.alt[i,j] = sum(temp.df$rel.freq[temp.df$pyr.sub==pyr.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

pyr.mat = as.matrix(pyr.sub.alt[1:5])
D2 = mahalanobis(pyr.mat,colMeans(pyr.mat),cov(pyr.mat))

pyr.obs = as.vector(pyr.freq.df$rel.freq)
names(pyr.obs) = pyr.freq.df$pyr.sub
D2.obs = mahalanobis(pyr.obs[1:5],colMeans(pyr.mat),cov(pyr.mat))

p.val = mean(D2>=D2.obs) #0.055

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=450,yend=150),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=550,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("Pyrmidine Substitution: Nonsyn Alternative Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(pyr.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(pyr.obs[1:5])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```

# Sense strand substitutions (12)

## Sense strand substitutions: null model - uniform distribution

```{r}

set.seed(5)
nsims = 1e4
sns.sub.null = as.data.frame(matrix(nrow=nsims,ncol=12))
colnames(sns.sub.null) = sns.subs

temp.df = tot.mut[,c("nuc.sub","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$nuc.sub = sample(sns.subs,nrow(temp.df),replace=T,prob=rep(1/12,12))
  
  for (j in 1:length(sns.subs)) {
    
    sns.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$nuc.sub==sns.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

sns.mat = as.matrix(sns.sub.null[1:11])
D2 = mahalanobis(sns.mat,colMeans(sns.mat),cov(sns.mat))

sns.obs = as.vector(sns.freq.df$rel.freq)
names(sns.obs) = sns.freq.df$sns.sub
D2.obs = mahalanobis(sns.obs[1:11],colMeans(sns.mat),cov(sns.mat))

p.val = mean(D2>=D2.obs) # 0.23

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=650,yend=500),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=750,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("Sense Strand Substitutions: Uniform Null Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(sns.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(sns.obs[1:11])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```

## Sense strand substitutions: null model - control ACGT content

```{r}

AR.ACGT.con = c(rep(0.224,3),rep(0.281,3),rep(0.288,3),rep(0.208,3))
ESR1.ACGT.con = c(rep(0.233,3),rep(0.288,3),rep(0.285,3),rep(0.194,3))
KIT.ACGT.con = c(rep(0.289,3),rep(0.218,3),rep(0.231,3),rep(0.262,3))



set.seed(5)
nsims = 1e4
sns.sub.null = as.data.frame(matrix(nrow=nsims,ncol=12))
colnames(sns.sub.null) = sns.subs

temp.df = tot.mut[,c("nuc.sub","rel.freq","gene")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$nuc.sub[temp.df$gene=="AR"] = sample(sns.subs,nrow(temp.df[temp.df$gene=="AR",]),replace=T,prob=AR.ACGT.con)
  temp.df$nuc.sub[temp.df$gene=="ESR1"] = sample(sns.subs,nrow(temp.df[temp.df$gene=="ESR1",]),replace=T,prob=ESR1.ACGT.con)
  temp.df$nuc.sub[temp.df$gene=="KIT"] = sample(sns.subs,nrow(temp.df[temp.df$gene=="KIT",]),replace=T,prob=KIT.ACGT.con)
  
  for (j in 1:length(sns.subs)) {
    
    sns.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$nuc.sub==sns.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

sns.mat = as.matrix(sns.sub.null[1:11])
D2 = mahalanobis(sns.mat,colMeans(sns.mat),cov(sns.mat))

sns.obs = as.vector(sns.freq.df$rel.freq)
names(sns.obs) = sns.freq.df$sns.sub
D2.obs = mahalanobis(sns.obs[1:11],colMeans(sns.mat),cov(sns.mat))

p.val = mean(D2>=D2.obs) # 0.17

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=600,yend=450),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=700,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("Sense Strand Substitution: ACGT Null Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(sns.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(sns.obs[1:11])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```

## Sense strand substitutions: null model - consider nonsynonymous mutations

```{r}

AR.prob = c()
ESR1.prob = c()
KIT.prob = c()

for (i in 1:length(sns.subs)) {
  AR.prob = c(AR.prob,mean(AR.nsyn.lim$bp.sub==sns.subs[i]))
  ESR1.prob = c(ESR1.prob,mean(ESR1.nsyn.lim$bp.sub==sns.subs[i]))
  KIT.prob = c(KIT.prob,mean(KIT.nsyn.lim$bp.sub==sns.subs[i]))
}

set.seed(5)
nsims = 1e4
sns.sub.null = as.data.frame(matrix(nrow=nsims,ncol=12))
colnames(sns.sub.null) = sns.subs

temp.df = tot.mut[,c("nuc.sub","rel.freq","gene")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$nuc.sub[temp.df$gene=="AR"] = sample(sns.subs,nrow(temp.df[temp.df$gene=="AR",]),replace=T,prob=AR.prob)
  temp.df$nuc.sub[temp.df$gene=="ESR1"] = sample(sns.subs,nrow(temp.df[temp.df$gene=="ESR1",]),replace=T,prob=ESR1.prob)
  temp.df$nuc.sub[temp.df$gene=="KIT"] = sample(sns.subs,nrow(temp.df[temp.df$gene=="KIT",]),replace=T,prob=KIT.prob)
  
  for (j in 1:length(sns.subs)) {
    
    sns.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$nuc.sub==sns.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

sns.mat = as.matrix(sns.sub.null[1:11])
D2 = mahalanobis(sns.mat,colMeans(sns.mat),cov(sns.mat))

sns.obs = as.vector(sns.freq.df$rel.freq)
names(sns.obs) = sns.freq.df$sns.sub
D2.obs = mahalanobis(sns.obs[1:11],colMeans(sns.mat),cov(sns.mat))

p.val = mean(D2>=D2.obs) # 0.17

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=600,yend=450),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=700,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("Sense Strand Substitution: Nonsyn Null Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(sns.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(sns.obs[1:11])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```



## Sense strand substitutions: alternative model

```{r}

```

# Trinucleotide pyrmidine substitutions (96)

## Trinucleotide pyrmidine substitutions: null model - uniform distribution

```{r}

set.seed(5)
nsims = 5e3
tri.sub.null = as.data.frame(matrix(nrow=nsims,ncol=96))
colnames(tri.sub.null) = tri.subs

temp.df = tot.mut[,c("tri.sub","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$tri.sub = sample(tri.subs,nrow(temp.df),replace=T,prob=rep(1/96,96))

  for (j in 1:length(tri.subs)) {
    
    tri.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$tri.sub==tri.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

tri.mat = as.matrix(tri.sub.null[1:95])
D2 = mahalanobis(tri.mat,colMeans(tri.mat),cov(tri.mat))

tri.obs = as.vector(tri.freq.df$rel.freq)
names(tri.obs) = tri.freq.df$tri.sub
D2.obs = mahalanobis(tri.obs[1:95],colMeans(tri.mat),cov(tri.mat))

p.val = mean(D2>=D2.obs) # 0.017

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=500,yend=50),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=600,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("Trinucleotide Pyrmidine Substitutions: Uniform Null Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(tri.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(tri.obs[1:95])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```

## Trinucleotide pyrmidine substitutions: control for nonsynonymous mutations

```{r}

AR.prob = c()
ESR1.prob = c()
KIT.prob = c()

for (i in 1:length(tri.subs)) {
  AR.prob = c(AR.prob,mean(AR.nsyn.lim$trinuc.pyr==tri.subs[i]))
  ESR1.prob = c(ESR1.prob,mean(ESR1.nsyn.lim$trinuc.pyr==tri.subs[i]))
  KIT.prob = c(KIT.prob,mean(KIT.nsyn.lim$trinuc.pyr==tri.subs[i]))
}

set.seed(5)
nsims = 5e3
tri.sub.null = as.data.frame(matrix(nrow=nsims,ncol=96))
colnames(tri.sub.null) = tri.subs

temp.df = tot.mut[,c("tri.sub","rel.freq","gene")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$tri.sub[temp.df$gene=="AR"] = sample(tri.subs,nrow(temp.df[temp.df$gene=="AR",]),replace=T,prob=AR.prob)
  temp.df$tri.sub[temp.df$gene=="ESR1"] = sample(tri.subs,nrow(temp.df[temp.df$gene=="ESR1",]),replace=T,prob=ESR1.prob)
  temp.df$tri.sub[temp.df$gene=="KIT"] = sample(tri.subs,nrow(temp.df[temp.df$gene=="KIT",]),replace=T,prob=KIT.prob)

  for (j in 1:length(tri.subs)) {
    
    tri.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$tri.sub==tri.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

tri.mat = as.matrix(tri.sub.null[1:95])
D2 = mahalanobis(tri.mat,colMeans(tri.mat),cov(tri.mat))

tri.obs = as.vector(tri.freq.df$rel.freq)
names(tri.obs) = tri.freq.df$tri.sub
D2.obs = mahalanobis(tri.obs[1:95],colMeans(tri.mat),cov(tri.mat))

p.val = mean(D2>=D2.obs) # 0.017

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=500,yend=50),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=600,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("Trinucleotide Pyrmidine Substitutions: Nonsyn Null Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(tri.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(tri.obs[1:95])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```

## Trinucleotide pyrmidine substitutions: alternative model

```{r}

set.seed(5)
nsims = 5e3
tri.sub.alt = as.data.frame(matrix(nrow=nsims,ncol=96))
colnames(tri.sub.alt) = tri.subs

temp.df = tot.mut[,c("tri.sub","rel.freq","gene")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$tri.sub[temp.df$gene=="AR"] = sample(names(PCa.sig),sum(temp.df$gene=="AR"),replace=T,prob=PCa.sig)
  temp.df$tri.sub[temp.df$gene=="ESR1"] = sample(names(BCa.sig),sum(temp.df$gene=="ESR1"),replace=T,prob=BCa.sig)
  temp.df$tri.sub[temp.df$gene=="KIT"] = sample(names(GIST.sig),sum(temp.df$gene=="KIT"),replace=T,prob=GIST.sig)

  for (j in 1:length(tri.subs)) {
    
    tri.sub.alt[i,j] = sum(temp.df$rel.freq[temp.df$tri.sub==tri.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

tri.mat = as.matrix(tri.sub.alt[1:95])
D2 = mahalanobis(tri.mat,colMeans(tri.mat),cov(tri.mat))

tri.obs = as.vector(tri.freq.df$rel.freq)
names(tri.obs) = tri.freq.df$tri.sub
D2.obs = mahalanobis(tri.obs[1:95],colMeans(tri.mat),cov(tri.mat))

p.val = mean(D2>=D2.obs) # 0.001

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=250,yend=50),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=350,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("Trinucleotide Pyrmidine Substitution: Alternative Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(tri.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(tri.obs[1:95])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )


```

## Trinucleotide pyrmidine substitutions: alternative model, consider nonsynonymous mutations

```{r}

AR.nsyn.tab = table(AR.nsyn.lim$trinuc.pyr)
AR.nsyn.tab = AR.nsyn.tab[tri.subs]
AR.nsyn.sig = PCa.sig*AR.nsyn.tab
AR.nsyn.sig = AR.nsyn.sig/sum(AR.nsyn.sig)

ESR1.nsyn.tab = table(ESR1.nsyn.lim$trinuc.pyr)
ESR1.nsyn.tab = ESR1.nsyn.tab[tri.subs]
ESR1.nsyn.sig = BCa.sig*ESR1.nsyn.tab
ESR1.nsyn.sig = ESR1.nsyn.sig/sum(ESR1.nsyn.sig)

KIT.nsyn.tab = table(KIT.nsyn.lim$trinuc.pyr)
KIT.nsyn.tab = KIT.nsyn.tab[tri.subs]
KIT.nsyn.sig = GIST.sig*KIT.nsyn.tab
KIT.nsyn.sig = KIT.nsyn.sig/sum(KIT.nsyn.sig)

set.seed(5)
nsims = 5e3
tri.sub.alt = as.data.frame(matrix(nrow=nsims,ncol=96))
colnames(tri.sub.alt) = tri.subs

temp.df = tot.mut[,c("tri.sub","rel.freq","gene")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$tri.sub[temp.df$gene=="AR"] = sample(names(PCa.sig),sum(temp.df$gene=="AR"),replace=T,prob=AR.nsyn.sig)
  temp.df$tri.sub[temp.df$gene=="ESR1"] = sample(names(BCa.sig),sum(temp.df$gene=="ESR1"),replace=T,prob=ESR1.nsyn.sig)
  temp.df$tri.sub[temp.df$gene=="KIT"] = sample(names(GIST.sig),sum(temp.df$gene=="KIT"),replace=T,prob=KIT.nsyn.sig)

  for (j in 1:length(tri.subs)) {
    
    tri.sub.alt[i,j] = sum(temp.df$rel.freq[temp.df$tri.sub==tri.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

tri.mat = as.matrix(tri.sub.alt[1:95])
D2 = mahalanobis(tri.mat,colMeans(tri.mat),cov(tri.mat))

tri.obs = as.vector(tri.freq.df$rel.freq)
names(tri.obs) = tri.freq.df$tri.sub
D2.obs = mahalanobis(tri.obs[1:95],colMeans(tri.mat),cov(tri.mat))

p.val = mean(D2>=D2.obs) # 0.001

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=250,yend=50),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=350,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("Trinucleotide Pyrmidine Substitution: Nonsyn Alternative Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(tri.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(tri.obs[1:95])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )


```

# Positive Control: ABL

## Import/QC Data

```{r}

ABL.df = read.csv("ABLSummary2.csv",stringsAsFactors=F)
ABL.df$gene = "ABL"
ABL.df = ABL.df[,c(ncol(ABL.df),1:(ncol(ABL.df)-1))]

ABL.df = ABL.df[ABL.df$mut.codon!="multsub",]
ABL.df$rel.freq = ABL.df$freq/sum(ABL.df$freq)

ABL.TsTv = ABL.df[ABL.df$class!="?",]

ABL.mut = ABL.df[ABL.df$mut.codon!="?",]
ABL.mut$tri.sub = paste(ABL.mut$init.3nuc,ABL.mut$pyr.sub,sep=".")
ABL.mut$init.pyr = substr(ABL.mut$pyr.sub,1,1)

```

## ABL Allele distributions

```{r}

# Transitions/transversions

ABL.TsTv.r = ABL.TsTv[order(ABL.TsTv$freq,decreasing=T),]
ABL.TsTv.r$x = 1:nrow(ABL.TsTv.r)

ggplot(ABL.TsTv.r,aes(x=x,y=rel.freq))+theme_bw()+
  geom_bar(aes(fill=class),stat="identity")+
  scale_x_continuous(labels=c())+
  scale_fill_discrete("Class",labels=c("Transition","Transversion"))+
  ggtitle("Resistance Allele Distribution:\nObserved")+
  xlab("Resistance Variants")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=18,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=16,face="bold"),
    legend.position=c(0.8,0.8),
    legend.background = element_rect(fill="gray85")
  )

# Pyrmidine substitutions

ABL.mut.r = ABL.mut[order(ABL.mut$freq,decreasing=T),]
ABL.mut.r$x = 1:nrow(ABL.mut.r)

ggplot(ABL.mut.r,aes(x=x,y=rel.freq))+theme_bw()+
  geom_bar(aes(fill=pyr.sub),stat="identity")+
  scale_x_continuous(labels=c())+
  scale_fill_discrete("Pyrmidine\nSubstitutions",labels=c("C>A","C>G","C>T","T>A","T>C","T>G"))+
  ggtitle("Resistance Allele Distribution:\nABL")+
  xlab("Resistance Variants")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=22,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=22,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=22,face="bold"),
    legend.position=c(0.6,0.6),
    legend.background = element_rect(fill=alpha("gray85",0.5))
  )+
  guides(
    fill=guide_legend(nrow=3)
  )

ggsave("ABLResDist.png",width=6.5,height=6)

# Sense strand substitutions

ggplot(ABL.mut.r,aes(x=x,y=rel.freq))+theme_bw()+
  geom_bar(aes(fill=nuc.sub),stat="identity")+
  scale_x_continuous(labels=c())+
  scale_fill_discrete("Sense Substitutions")+
  ggtitle("Resistance Allele Distribution:\nObserved")+
  xlab("Resistance Variants")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=18,face="bold"),
    legend.title.align=0.5,
    legend.position=c(0.6,0.75),
    legend.text=element_text(size=16,face="bold"),
    legend.background = element_rect(fill="gray85")
  )+
  guides(fill=guide_legend(nrow=3))

# Trinucleotide context

ggplot(ABL.mut.r,aes(x=x,y=rel.freq))+theme_bw()+
  geom_bar(aes(fill=pyr.sub,color=init.3nuc),stat="identity")+
  scale_x_continuous(labels=c())+
  scale_fill_discrete("Pyrmidine Substitution")+
  scale_color_discrete("Trinucleotide Context")+
  ggtitle("Resistance Allele Distribution:\nObserved")+
  xlab("Resistance Variants")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=12,face="bold"),
    legend.title.align=0.5,
    legend.position=c(0.6,0.6),
    legend.text=element_text(size=10,face="bold"),
    legend.background = element_rect(fill="gray85")
  )+
  guides(
    fill=guide_legend(nrow=2),
    color=guide_legend(nrow=5)
  )


```

```{r}

# Transitions/transversions

Ts.freq = sum(ABL.TsTv$rel.freq[ABL.TsTv$class=="Ts"])
Tv.freq = sum(ABL.TsTv$rel.freq[ABL.TsTv$class=="Tv"])

TsTv.freq = data.frame(class=c("Ts","Tv"),rel.freq=c(Ts.freq,Tv.freq))
ggplot(TsTv.freq,aes(x=class,y=rel.freq))+theme_bw()+
  geom_bar(aes(fill=class),stat="identity")+
  ggtitle("Transition/Transversion Mutational Signature")+
  scale_x_discrete(labels=c("Transitions","Transversions"))+
  xlab("")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=18,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=16,face="bold"),
    legend.position=c(0.8,0.8),
    legend.background = element_rect(fill="gray85")
  )+
  guides(fill=F)

## Pyrmidine substitutions

pyr.tab = table(ABL.mut$pyr.sub)
pyr.subs = names(pyr.tab)
pyr.freq = pyr.tab
for (i in 1:length(pyr.tab)) {
  pyr.freq[i] = sum(ABL.mut$rel.freq[ABL.mut$pyr.sub==names(pyr.tab)[i]])
}

pyr.freq.df = as.data.frame(pyr.freq)
colnames(pyr.freq.df) = c("pyr.sub","rel.freq")

ggplot(pyr.freq.df,aes(x=pyr.sub,y=rel.freq))+theme_bw()+
  geom_hline(yintercept=1/6,color="gray",linetype="dashed",size=1.5)+
  geom_bar(aes(fill=pyr.sub),stat="identity")+
  ggtitle("Pyrmidine Mutational Signature")+
  scale_x_discrete(labels=pyr.subs)+
  xlab("")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=18,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=16,face="bold"),
    legend.position=c(0.8,0.8),
    legend.background = element_rect(fill="gray85")
  )+
  guides(fill=F)

## Sense strand substitutions

sns.tab = table(ABL.mut$sns.sub)
# fill in zero counts
sns.tab = sns.tab[sns.subs]
names(sns.tab) = sns.subs
sns.tab[is.na(sns.tab)] = 0

sns.freq = sns.tab
for (i in 1:length(sns.tab)) {
  sns.freq[i] = sum(ABL.mut$rel.freq[ABL.mut$nuc.sub==names(sns.tab)[i]])
}

sns.freq.df = as.data.frame(sns.freq)
colnames(sns.freq.df) = c("sns.sub","rel.freq")

ggplot(sns.freq.df,aes(x=sns.sub,y=rel.freq))+theme_bw()+
  geom_hline(yintercept=1/12,color="gray",linetype="dashed",size=1.5)+
  geom_bar(aes(fill=sns.sub),stat="identity")+
  ggtitle("Sense Strand Mutational Signature")+
  scale_x_discrete(labels=sns.subs)+
  xlab("")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=18,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=16,face="bold"),
    legend.position=c(0.8,0.8),
    legend.background = element_rect(fill="gray85")
  )+
  guides(fill=F)

## Trinucleotide

tri.tab = table(ABL.mut$tri.sub)
# fill in zero counts
tri.tab = tri.tab[tri.subs]
names(tri.tab) = tri.subs
tri.tab[is.na(tri.tab)] = 0

tri.freq = tri.tab
for (i in 1:length(tri.tab)) {
  tri.freq[i] = sum(ABL.mut$rel.freq[ABL.mut$tri.sub==names(tri.tab)[i]])
}

tri.freq.df = as.data.frame(tri.freq)
colnames(tri.freq.df) = c("tri.sub","rel.freq")
tri.freq.df$init.3nuc = substr(tri.freq.df$tri.sub,1,3)
tri.freq.df$init.pyr = substr(tri.freq.df$tri.sub,2,2)
tri.freq.df$pyr.sub = substr(tri.freq.df$tri.sub,5,6)

ggplot()+theme_bw()+
  geom_hline(yintercept=1/96,color="gray",linetype="dashed",size=1.5)+
  geom_bar(data=tri.freq.df[tri.freq.df$init.pyr=="C",],aes(x=init.3nuc,y=rel.freq,fill=pyr.sub),stat="identity")+
  facet_grid(.~pyr.sub)+
  # scale_x_discrete(labels=c())+
  ggtitle("Trinucleotide Mutational Signature")+
  # scale_x_discrete(labels=tri.nucs)+
  xlab("")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=14),
    axis.text.x=element_text(angle=90),
    legend.title=element_text(size=18,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=16,face="bold"),
    # strip.background=element_blank(),
    strip.text=element_text(size=12,face="bold")
  )+
  guides(fill=F)

ggplot()+theme_bw()+
  geom_hline(yintercept=1/96,color="gray",linetype="dashed",size=1.5)+
  geom_bar(data=tri.freq.df[tri.freq.df$init.pyr=="T",],aes(x=init.3nuc,y=rel.freq,fill=pyr.sub),stat="identity")+
  facet_grid(.~pyr.sub)+
  # scale_x_discrete(labels=c())+
  ggtitle("Trinucleotide Mutational Signature")+
  # scale_x_discrete(labels=tri.nucs)+
  xlab("")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=14),
    axis.text.x=element_text(angle=90),
    legend.title=element_text(size=18,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=16,face="bold"),
    # strip.background=element_blank(),
    strip.text=element_text(size=12,face="bold")
  )+
  guides(fill=F)

```

## ABL: Mutational signature for alternative models:

```{r}

CML.pyr.sig = c(4+6,6+6,88+68,1+2,18+17,3+5) # from Figure 1
names(CML.pyr.sig) = pyr.subs

```

## ABL: Transition/transversion - Ts:Tv ratio = 0.5

```{r}

set.seed(5)
nsims = 1e4
TsTv.null = as.data.frame(matrix(nrow=nsims,ncol=2))
colnames(TsTv.null) = c("Ts","Tv")

temp.df = ABL.TsTv[,c("class","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$class = sample(c("Ts","Tv"),nrow(temp.df),replace=T,prob=c(1/3,2/3))
  
  TsTv.null$Ts[i] = sum(temp.df$rel.freq[temp.df$class=="Ts"])
  TsTv.null$Tv[i] = sum(temp.df$rel.freq[temp.df$class!="Ts"])
  
}

Ts.freq = sum(ABL.TsTv$rel.freq[ABL.TsTv$class=="Ts"])
p.val = mean(TsTv.null$Ts>=Ts.freq) # 0.002

ggplot() + theme_bw()+
  geom_histogram(data=TsTv.null,aes(x=Ts),fill="#4286f4")+
  geom_segment(aes(x=Ts.freq,xend=Ts.freq,y=500,yend=90),color="red",arrow=arrow(length=unit(0.5,"cm")),size=2)+
  geom_text(aes(x=Ts.freq,y=600,label=paste("Observed: ",round(Ts.freq,2),"\np-value=",round(p.val,3),sep="")),size=6,fontface="bold",color="red")+
  ggtitle("ABL Transition Distribution: Uniform Null Model")+
  xlab("Transition Frequency")+ylab("")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```

## ABL Transitions/transversions: null model - conserve transition path frequency

```{r}

n.Ts.path = sum(ABL.TsTv$class=="Ts")
n.Tv.path = sum(ABL.TsTv$class=="Tv")

set.seed(5)
nsims = 1e4
TsTv.null = as.data.frame(matrix(nrow=nsims,ncol=2))
colnames(TsTv.null) = c("Ts","Tv")

temp.df = ABL.TsTv[,c("class","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$class = sample(c("Ts","Tv"),nrow(temp.df),replace=T,prob=c(n.Ts.path,n.Tv.path))
  
  TsTv.null$Ts[i] = sum(temp.df$rel.freq[temp.df$class=="Ts"])
  TsTv.null$Tv[i] = sum(temp.df$rel.freq[temp.df$class!="Ts"])
  
}

Ts.freq = sum(ABL.TsTv$rel.freq[ABL.TsTv$class=="Ts"])
p.val = mean(TsTv.null$Ts>=Ts.freq) # 0.006

ggplot() + theme_bw()+
  geom_histogram(data=TsTv.null,aes(x=Ts),fill="#4286f4")+
  geom_segment(aes(x=Ts.freq,xend=Ts.freq,y=500,yend=300),color="red",arrow=arrow(length=unit(0.5,"cm")),size=2)+
  geom_text(aes(x=Ts.freq,y=600,label=paste("Observed: ",round(Ts.freq,2),"\np-value=",round(p.val,3),sep="")),size=6,fontface="bold",color="red")+
  ggtitle("ABL Transition Distribution: Conserved Path Model")+
  xlab("Transition Frequency")+ylab("")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )


```

## ABL: Transition/transversion - alternative model

```{r}

n.Ts.alt = sum(CML.pyr.sig[names(CML.pyr.sig)%in%c("CT","TC")])
n.Tv.alt = sum(CML.pyr.sig[!names(CML.pyr.sig)%in%c("CT","TC")])


set.seed(5)
nsims = 1e4
TsTv.alt = as.data.frame(matrix(nrow=nsims,ncol=2))
colnames(TsTv.alt) = c("Ts","Tv")

temp.df = ABL.TsTv[,c("class","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$class = sample(c("Ts","Tv"),nrow(temp.df),replace=T,prob=c(n.Ts.alt,n.Tv.alt))
  
  TsTv.alt$Ts[i] = sum(temp.df$rel.freq[temp.df$class=="Ts"])
  TsTv.alt$Tv[i] = sum(temp.df$rel.freq[temp.df$class!="Ts"])
  
}

Ts.freq = sum(ABL.TsTv$rel.freq[ABL.TsTv$class=="Ts"])
p.val = mean(TsTv.alt$Ts>=Ts.freq) # 0.83

ggplot() + theme_bw()+
  geom_histogram(data=TsTv.alt,aes(x=Ts),fill="#4286f4")+
  geom_segment(aes(x=Ts.freq,xend=Ts.freq,y=500,yend=300),color="red",arrow=arrow(length=unit(0.5,"cm")),size=2)+
  geom_text(aes(x=Ts.freq,y=600,label=paste("Observed: ",round(Ts.freq,2),"\np-value=",round(p.val,3),sep="")),size=6,fontface="bold",color="red")+
  ggtitle("ABL Transition Distribution: Alternative Model")+
  xlab("Transition Frequency")+ylab("")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )


```

## ABL: Pyrmidine substitutions - null uniform distribution

```{r}

set.seed(5)
nsims = 1e4
pyr.sub.null = as.data.frame(matrix(nrow=nsims,ncol=6))
colnames(pyr.sub.null) = pyr.subs

temp.df = ABL.mut[,c("pyr.sub","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$pyr.sub = sample(pyr.subs,nrow(temp.df),replace=T,prob=rep(1/6,6))
  
  for (j in 1:length(pyr.subs)) {
    
    pyr.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$pyr.sub==pyr.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

pyr.mat = as.matrix(pyr.sub.null[1:5])
D2 = mahalanobis(pyr.mat,colMeans(pyr.mat),cov(pyr.mat))

pyr.obs = as.vector(pyr.freq.df$rel.freq)
names(pyr.obs) = pyr.freq.df$pyr.sub
D2.obs = mahalanobis(pyr.obs[1:5],colMeans(pyr.mat),cov(pyr.mat))

p.val = mean(D2>=D2.obs) #0.055

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=450,yend=150),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=550,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("ABL Pyrmidine Substitution: Uniform Null Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(pyr.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(pyr.obs[1:5])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```





## ABL: Pyrmidine substitutions - conserve path frequency

```{r}

n.path.pyr = table(ABL.mut$pyr.sub)
n.path.pyr = n.path.pyr[pyr.subs]
n.path.pyr[is.na(n.path.pyr)] = 0
names(n.path.pyr) = pyr.subs

set.seed(5)
nsims = 1e4
pyr.sub.null = as.data.frame(matrix(nrow=nsims,ncol=6))
colnames(pyr.sub.null) = pyr.subs

temp.df = ABL.mut[,c("pyr.sub","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$pyr.sub = sample(pyr.subs,nrow(temp.df),replace=T,prob=n.path.pyr)
  
  for (j in 1:length(pyr.subs)) {
    
    pyr.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$pyr.sub==pyr.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

pyr.mat = as.matrix(pyr.sub.null)
pyr.mat.f = pyr.mat[,colMeans(pyr.mat)!=0]
pyr.mat.f = pyr.mat.f[,1:(ncol(pyr.mat.f)-1)]
D2 = mahalanobis(pyr.mat.f,colMeans(pyr.mat.f),cov(pyr.mat.f))

pyr.obs = as.vector(pyr.freq.df$rel.freq)
names(pyr.obs) = pyr.freq.df$pyr.sub
pyr.obs.f = pyr.obs[colMeans(pyr.mat)!=0]
pyr.obs.f = pyr.obs.f[1:(length(pyr.obs.f)-1)]
D2.obs = mahalanobis(pyr.obs.f,colMeans(pyr.mat.f),cov(pyr.mat.f))

p.val = mean(D2>=D2.obs) #0.055

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=450,yend=150),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=550,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("ABL Pyrmidine Substitution: Conserved Path Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(pyr.mat.f,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(pyr.obs.f)))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```

## ABL: Pyrmidine substitutions - control for GC content

```{r}

ABL.GC.con = 0.59

set.seed(5)
nsims = 1e4
pyr.sub.null = as.data.frame(matrix(nrow=nsims,ncol=6))
colnames(pyr.sub.null) = pyr.subs

temp.df = ABL.mut[,c("pyr.sub","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$pyr.sub = sample(pyr.subs,nrow(temp.df),replace=T,prob=c(rep(ABL.GC.con,3),rep(1-ABL.GC.con,3)))
  
  for (j in 1:length(pyr.subs)) {
    
    pyr.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$pyr.sub==pyr.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

pyr.mat = as.matrix(pyr.sub.null)
pyr.mat.f = pyr.mat[,colMeans(pyr.mat)!=0]
pyr.mat.f = pyr.mat.f[,1:(ncol(pyr.mat.f)-1)]
D2 = mahalanobis(pyr.mat.f,colMeans(pyr.mat.f),cov(pyr.mat.f))

pyr.obs = as.vector(pyr.freq.df$rel.freq)
names(pyr.obs) = pyr.freq.df$pyr.sub
pyr.obs.f = pyr.obs[colMeans(pyr.mat)!=0]
pyr.obs.f = pyr.obs.f[1:(length(pyr.obs.f)-1)]
D2.obs = mahalanobis(pyr.obs.f,colMeans(pyr.mat.f),cov(pyr.mat.f))

p.val = mean(D2>=D2.obs) #0.055
ABL.pval = p.val

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=450,yend=150),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=550,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("ABL Pyrmidine Substitution: Null GC Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(pyr.mat.f,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(pyr.obs.f)))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```

## ABL: Pyrmidine substitutions - control for amino acid substitutions

```{r}

set.seed(5)
nsims = 1e4
pyr.sub.null = as.data.frame(matrix(nrow=nsims,ncol=6))
colnames(pyr.sub.null) = pyr.subs

temp.df = ABL.mut[,c("pyr.sub","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$pyr.sub = sample(pyr.subs,nrow(temp.df),replace=T,prob=c(68,76,58,64,58,68))
  
  for (j in 1:length(pyr.subs)) {
    
    pyr.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$pyr.sub==pyr.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

pyr.mat = as.matrix(pyr.sub.null[1:5])
D2 = mahalanobis(pyr.mat,colMeans(pyr.mat),cov(pyr.mat))

pyr.obs = as.vector(pyr.freq.df$rel.freq)
names(pyr.obs) = pyr.freq.df$pyr.sub
D2.obs = mahalanobis(pyr.obs[1:5],colMeans(pyr.mat),cov(pyr.mat))

p.val = mean(D2>=D2.obs) #0.055

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=525,yend=125),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=625,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=8,fontface="bold",color="red")+
  ggtitle("Null Model Pyrmidine Substitution:\nABL")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=22,face="bold"),
    axis.text=element_text(size=16)
  )

ggsave("ABLMahalanobisHist.png",width=6.5,height=6)

pca = prcomp(pyr.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(pyr.obs[1:5])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

write.table(pca.df,"ABLPCA.csv",row.names=F,col.names=F,sep=",")
write.table(pca.obs,"ABLPCAobs.csv",row.names=F,col.names=F,sep=",")

```

## ABL: Pyrmidine substitutions - alternative model

```{r}


set.seed(5)
nsims = 1e4
pyr.sub.alt = as.data.frame(matrix(nrow=nsims,ncol=6))
colnames(pyr.sub.alt) = pyr.subs

temp.df = ABL.mut[,c("pyr.sub","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$pyr.sub = sample(pyr.subs,nrow(temp.df),replace=T,prob=CML.pyr.sig)
  
  for (j in 1:length(pyr.subs)) {
    
    pyr.sub.alt[i,j] = sum(temp.df$rel.freq[temp.df$pyr.sub==pyr.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

pyr.mat = as.matrix(pyr.sub.alt)
pyr.mat.f = pyr.mat[,colMeans(pyr.mat)!=0]
pyr.mat.f = pyr.mat.f[,1:(ncol(pyr.mat.f)-1)]
D2 = mahalanobis(pyr.mat.f,colMeans(pyr.mat.f),cov(pyr.mat.f))

pyr.obs = as.vector(pyr.freq.df$rel.freq)
names(pyr.obs) = pyr.freq.df$pyr.sub
pyr.obs.f = pyr.obs[colMeans(pyr.mat)!=0]
pyr.obs.f = pyr.obs.f[1:(length(pyr.obs.f)-1)]
D2.obs = mahalanobis(pyr.obs.f,colMeans(pyr.mat.f),cov(pyr.mat.f))

p.val = mean(D2>=D2.obs) #0.055

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=450,yend=150),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=550,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("ABL Pyrmidine Substitution: Conserved Path Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(pyr.mat.f,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(pyr.obs.f)))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```

## ABL: Sense strand substitution: uniform null model

```{r}

set.seed(5)
nsims = 1e4
sns.sub.null = as.data.frame(matrix(nrow=nsims,ncol=12))
colnames(sns.sub.null) = sns.subs

temp.df = ABL.mut[,c("nuc.sub","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$nuc.sub = sample(sns.subs,nrow(temp.df),replace=T,prob=rep(1/12,12))
  
  for (j in 1:length(sns.subs)) {
    
    sns.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$nuc.sub==sns.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

sns.mat = as.matrix(sns.sub.null[1:11])
D2 = mahalanobis(sns.mat,colMeans(sns.mat),cov(sns.mat))

sns.obs = as.vector(sns.freq.df$rel.freq)
names(sns.obs) = sns.freq.df$sns.sub
D2.obs = mahalanobis(sns.obs[1:11],colMeans(sns.mat),cov(sns.mat))

p.val = mean(D2>=D2.obs) # 0.392

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=650,yend=500),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=750,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("ABL Sense Strand Substitutions: Uniform Null Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(sns.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(sns.obs[1:11])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```



## ABL: Sense strand substitution: conserve path model

```{r}

n.path.sns = table(ABL.mut$nuc.sub)
n.path.sns = n.path.sns[sns.subs]
n.path.sns[is.na(n.path.sns)] = 0
names(n.path.sns) = sns.subs

set.seed(5)
nsims = 1e4
sns.sub.null = as.data.frame(matrix(nrow=nsims,ncol=12))
colnames(sns.sub.null) = sns.subs

temp.df = ABL.mut[,c("nuc.sub","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$nuc.sub = sample(sns.subs,nrow(temp.df),replace=T,prob=n.path.sns)
  
  for (j in 1:length(sns.subs)) {
    
    sns.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$nuc.sub==sns.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

sns.mat = as.matrix(sns.sub.null)
sns.mat.f = sns.mat[,colMeans(sns.mat)!=0]
sns.mat.f = sns.mat.f[,1:(ncol(sns.mat.f)-1)]
D2 = mahalanobis(sns.mat.f,colMeans(sns.mat.f),cov(sns.mat.f))

sns.obs = as.vector(sns.freq.df$rel.freq)
names(sns.obs) = sns.freq.df$sns.sub
sns.obs.f = sns.obs[colMeans(sns.mat)!=0]
sns.obs.f = sns.obs.f[1:(length(sns.obs.f)-1)]
D2.obs = mahalanobis(sns.obs.f,colMeans(sns.mat.f),cov(sns.mat.f))

p.val = mean(D2>=D2.obs) # 0.392

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=650,yend=500),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=750,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("ABL Sense Strand Substitutions: Conserved Path Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(sns.mat.f,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(sns.obs.f)))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```

## ABL Trinucleotide pyrmidine substitutions: null model - uniform distribution

```{r}

set.seed(5)
nsims = 5e3
tri.sub.null = as.data.frame(matrix(nrow=nsims,ncol=96))
colnames(tri.sub.null) = tri.subs

temp.df = ABL.mut[,c("tri.sub","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$tri.sub = sample(tri.subs,nrow(temp.df),replace=T,prob=rep(1/96,96))

  for (j in 1:length(tri.subs)) {
    
    tri.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$tri.sub==tri.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

tri.mat = as.matrix(tri.sub.null[1:95])
D2 = mahalanobis(tri.mat,colMeans(tri.mat),cov(tri.mat))

tri.obs = as.vector(tri.freq.df$rel.freq)
names(tri.obs) = tri.freq.df$tri.sub
D2.obs = mahalanobis(tri.obs[1:95],colMeans(tri.mat),cov(tri.mat))

p.val = mean(D2>=D2.obs) # 0.087

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=500,yend=50),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=600,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("ABL Trinucleotide Pyrmidine Substitutions: Uniform Null Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(tri.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(tri.obs[1:95])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```

## ABL Trinucleotide pyrmidine substitutions: conserved path model

```{r}

n.path.tri = table(ABL.mut$tri.sub)
n.path.tri = n.path.tri[tri.subs]
n.path.tri[is.na(n.path.tri)] = 0
names(n.path.tri) = tri.subs

set.seed(5)
nsims = 5e3
tri.sub.null = as.data.frame(matrix(nrow=nsims,ncol=96))
colnames(tri.sub.null) = tri.subs

temp.df = ABL.mut[,c("tri.sub","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$tri.sub = sample(tri.subs,nrow(temp.df),replace=T,prob=n.path.tri)

  for (j in 1:length(tri.subs)) {
    
    tri.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$tri.sub==tri.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

tri.mat = as.matrix(tri.sub.null)
tri.mat.f = tri.mat[,colMeans(tri.mat)!=0]
tri.mat.f = tri.mat.f[,1:(ncol(tri.mat.f)-1)]
D2 = mahalanobis(tri.mat.f,colMeans(tri.mat.f),cov(tri.mat.f))

tri.obs = as.vector(tri.freq.df$rel.freq)
names(tri.obs) = tri.freq.df$tri.sub
tri.obs.f = tri.obs[colMeans(tri.mat)!=0]
tri.obs.f = tri.obs.f[1:(length(tri.obs.f)-1)]
D2.obs = mahalanobis(tri.obs.f,colMeans(tri.mat.f),cov(tri.mat.f))


p.val = mean(D2>=D2.obs) # 0.087

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=500,yend=50),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=600,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("ABL Trinucleotide Pyrmidine Substitutions: Conserved Path Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(tri.mat.f,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(tri.obs.f)))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```



# Negative Control: EGFR

## Import/QC Data

```{r}

EGFR.df = read.csv("EGFRSummary.csv",stringsAsFactors=F)
EGFR.df$gene = "EGFR"
EGFR.df = EGFR.df[,c(ncol(EGFR.df),1:(ncol(EGFR.df)-1))]

EGFR.df = EGFR.df[EGFR.df$mut.codon!="multsub",]
EGFR.df$rel.freq = EGFR.df$freq/sum(EGFR.df$freq)

EGFR.TsTv = EGFR.df[EGFR.df$class!="?",]

EGFR.mut = EGFR.df[EGFR.df$mut.codon!="?",]
EGFR.mut$tri.sub = paste(EGFR.mut$init.3nuc,EGFR.mut$pyr.sub,sep=".")
EGFR.mut$init.pyr = substr(EGFR.mut$pyr.sub,1,1)

```

## EGFR Allele distributions

```{r}

# Transitions/transversions

EGFR.TsTv.r = EGFR.TsTv[order(EGFR.TsTv$freq,decreasing=T),]
EGFR.TsTv.r$x = 1:nrow(EGFR.TsTv.r)

ggplot(EGFR.TsTv.r,aes(x=x,y=rel.freq))+theme_bw()+
  geom_bar(aes(fill=class),stat="identity")+
  scale_x_continuous(labels=c())+
  scale_fill_discrete("Class",labels=c("Transition","Transversion"))+
  ggtitle("Resistance Allele Distribution:\nObserved")+
  xlab("Resistance Variants")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=18,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=16,face="bold"),
    legend.position=c(0.8,0.8),
    legend.background = element_rect(fill="gray85")
  )

# Pyrmidine substitutions

EGFR.mut.r = EGFR.mut[order(EGFR.mut$freq,decreasing=T),]
EGFR.mut.r$x = 1:nrow(EGFR.mut.r)
EGFR.mut.r$pyr.sub = factor(EGFR.mut.r$pyr.sub,levels=pyr.subs)

ggplot(EGFR.mut.r,aes(x=x,y=rel.freq))+theme_bw()+
  geom_bar(aes(fill=pyr.sub),stat="identity")+
  scale_x_continuous(labels=c())+
  scale_fill_discrete("Pyrmidine\nSubstitutions",labels=c("C>A","C>G","C>T","T>A","T>C","T>G"),drop=F)+
  ggtitle("Resistance Allele Distribution:\nEGFR")+
  xlab("Resistance Variants")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=22,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=22,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=22,face="bold"),
    legend.position=c(0.6,0.6),
    legend.background = element_rect(fill=alpha("gray85",0.5))
  )+
  guides(
    fill=guide_legend(nrow=3)
  )

ggsave("EGFRResDist.png",width=6.5,height=6)

# Sense strand substitutions

ggplot(EGFR.mut.r,aes(x=x,y=rel.freq))+theme_bw()+
  geom_bar(aes(fill=nuc.sub),stat="identity")+
  scale_x_continuous(labels=c())+
  scale_fill_discrete("Sense Substitutions")+
  ggtitle("Resistance Allele Distribution:\nObserved")+
  xlab("Resistance Variants")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=18,face="bold"),
    legend.title.align=0.5,
    legend.position=c(0.6,0.75),
    legend.text=element_text(size=16,face="bold"),
    legend.background = element_rect(fill="gray85")
  )+
  guides(fill=guide_legend(nrow=3))

# Trinucleotide context

ggplot(EGFR.mut.r,aes(x=x,y=rel.freq))+theme_bw()+
  geom_bar(aes(fill=pyr.sub,color=init.3nuc),stat="identity")+
  scale_x_continuous(labels=c())+
  scale_fill_discrete("Pyrmidine Substitution")+
  scale_color_discrete("Trinucleotide Context")+
  ggtitle("Resistance Allele Distribution:\nObserved")+
  xlab("Resistance Variants")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=12,face="bold"),
    legend.title.align=0.5,
    legend.position=c(0.6,0.6),
    legend.text=element_text(size=10,face="bold"),
    legend.background = element_rect(fill="gray85")
  )+
  guides(
    fill=guide_legend(nrow=2),
    color=guide_legend(nrow=5)
  )


```

```{r}

# Transitions/transversions

Ts.freq = sum(EGFR.TsTv$rel.freq[EGFR.TsTv$class=="Ts"])
Tv.freq = sum(EGFR.TsTv$rel.freq[EGFR.TsTv$class=="Tv"])

TsTv.freq = data.frame(class=c("Ts","Tv"),rel.freq=c(Ts.freq,Tv.freq))
ggplot(TsTv.freq,aes(x=class,y=rel.freq))+theme_bw()+
  geom_bar(aes(fill=class),stat="identity")+
  ggtitle("Transition/Transversion Mutational Signature")+
  scale_x_discrete(labels=c("Transitions","Transversions"))+
  xlab("")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=18,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=16,face="bold"),
    legend.position=c(0.8,0.8),
    legend.background = element_rect(fill="gray85")
  )+
  guides(fill=F)

## Pyrmidine substitutions

pyr.tab = table(EGFR.mut$pyr.sub)
# fill in zero counts
pyr.tab = pyr.tab[pyr.subs]
names(pyr.tab) = pyr.subs
pyr.tab[is.na(pyr.tab)] = 0

pyr.freq = pyr.tab
for (i in 1:length(pyr.tab)) {
  pyr.freq[i] = sum(EGFR.mut$rel.freq[EGFR.mut$pyr.sub==names(pyr.tab)[i]])
}

pyr.freq.df = as.data.frame(pyr.freq)
colnames(pyr.freq.df) = c("pyr.sub","rel.freq")

ggplot(pyr.freq.df,aes(x=pyr.sub,y=rel.freq))+theme_bw()+
  geom_hline(yintercept=1/6,color="gray",linetype="dashed",size=1.5)+
  geom_bar(aes(fill=pyr.sub),stat="identity")+
  ggtitle("Pyrmidine Mutational Signature")+
  scale_x_discrete(labels=pyr.subs)+
  xlab("")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=18,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=16,face="bold"),
    legend.position=c(0.8,0.8),
    legend.background = element_rect(fill="gray85")
  )+
  guides(fill=F)

## Sense strand substitutions

sns.tab = table(EGFR.mut$sns.sub)
# fill in zero counts
sns.tab = sns.tab[sns.subs]
names(sns.tab) = sns.subs
sns.tab[is.na(sns.tab)] = 0

sns.freq = sns.tab
for (i in 1:length(sns.tab)) {
  sns.freq[i] = sum(EGFR.mut$rel.freq[EGFR.mut$nuc.sub==names(sns.tab)[i]])
}

sns.freq.df = as.data.frame(sns.freq)
colnames(sns.freq.df) = c("sns.sub","rel.freq")

ggplot(sns.freq.df,aes(x=sns.sub,y=rel.freq))+theme_bw()+
  geom_hline(yintercept=1/12,color="gray",linetype="dashed",size=1.5)+
  geom_bar(aes(fill=sns.sub),stat="identity")+
  ggtitle("Sense Strand Mutational Signature")+
  scale_x_discrete(labels=sns.subs)+
  xlab("")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16),
    legend.title=element_text(size=18,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=16,face="bold"),
    legend.position=c(0.8,0.8),
    legend.background = element_rect(fill="gray85")
  )+
  guides(fill=F)

## Trinucleotide

tri.tab = table(EGFR.mut$tri.sub)
# fill in zero counts
tri.tab = tri.tab[tri.subs]
names(tri.tab) = tri.subs
tri.tab[is.na(tri.tab)] = 0

tri.freq = tri.tab
for (i in 1:length(tri.tab)) {
  tri.freq[i] = sum(EGFR.mut$rel.freq[EGFR.mut$tri.sub==names(tri.tab)[i]])
}

tri.freq.df = as.data.frame(tri.freq)
colnames(tri.freq.df) = c("tri.sub","rel.freq")
tri.freq.df$init.3nuc = substr(tri.freq.df$tri.sub,1,3)
tri.freq.df$init.pyr = substr(tri.freq.df$tri.sub,2,2)
tri.freq.df$pyr.sub = substr(tri.freq.df$tri.sub,5,6)

ggplot()+theme_bw()+
  geom_hline(yintercept=1/96,color="gray",linetype="dashed",size=1.5)+
  geom_bar(data=tri.freq.df[tri.freq.df$init.pyr=="C",],aes(x=init.3nuc,y=rel.freq,fill=pyr.sub),stat="identity")+
  facet_grid(.~pyr.sub)+
  # scale_x_discrete(labels=c())+
  ggtitle("Trinucleotide Mutational Signature")+
  # scale_x_discrete(labels=tri.nucs)+
  xlab("")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=14),
    axis.text.x=element_text(angle=90),
    legend.title=element_text(size=18,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=16,face="bold"),
    # strip.background=element_blank(),
    strip.text=element_text(size=12,face="bold")
  )+
  guides(fill=F)

ggplot()+theme_bw()+
  geom_hline(yintercept=1/96,color="gray",linetype="dashed",size=1.5)+
  geom_bar(data=tri.freq.df[tri.freq.df$init.pyr=="T",],aes(x=init.3nuc,y=rel.freq,fill=pyr.sub),stat="identity")+
  facet_grid(.~pyr.sub)+
  # scale_x_discrete(labels=c())+
  ggtitle("Trinucleotide Mutational Signature")+
  # scale_x_discrete(labels=tri.nucs)+
  xlab("")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=14),
    axis.text.x=element_text(angle=90),
    legend.title=element_text(size=18,face="bold"),
    legend.title.align=0.5,
    legend.text=element_text(size=16,face="bold"),
    # strip.background=element_blank(),
    strip.text=element_text(size=12,face="bold")
  )+
  guides(fill=F)

```

## EGFR: Mutational signature for alternative models:

```{r}

# Lung cancer
LUAD.df = cncr.types[cncr.types$Cancer.Types=="Breast-cancer",]
LUAD.conv = colMeans(LUAD.df[4:ncol(LUAD.df)])
LUAD.conv = LUAD.conv/sum(LUAD.conv)
LUAD.sig = as.vector(as.matrix(mut.sig.df[3:ncol(mut.sig.df)]) %*% LUAD.conv)
names(LUAD.sig) = paste(mut.sig.df$SubType,".",substr(mut.sig.df$Type,1,1),substr(mut.sig.df$Type,3,3),sep="")
LUAD.sig = LUAD.sig[tri.subs]

LUAD.pyr.sig = as.table(matrix(nrow=1,ncol=6))
names(LUAD.pyr.sig) = pyr.subs
for (i in 1:length(pyr.subs)) {
  LUAD.pyr.sig[,i] = sum(LUAD.sig[substr(names(LUAD.sig),5,6)==pyr.subs[i]])
}

```

## EGFR: Transition/transversion - Ts:Tv ratio = 0.5
## EGFR Transition/transversion - null uniform distribution
```{r}

set.seed(5)
nsims = 1e4
TsTv.null = as.data.frame(matrix(nrow=nsims,ncol=2))
colnames(TsTv.null) = c("Ts","Tv")

temp.df = EGFR.TsTv[,c("class","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$class = sample(c("Ts","Tv"),nrow(temp.df),replace=T,prob=c(1/3,2/3))
  
  TsTv.null$Ts[i] = sum(temp.df$rel.freq[temp.df$class=="Ts"])
  TsTv.null$Tv[i] = sum(temp.df$rel.freq[temp.df$class!="Ts"])
  
}

Ts.freq = sum(EGFR.TsTv$rel.freq[EGFR.TsTv$class=="Ts"])
p.val = mean(TsTv.null$Ts>=Ts.freq) # 0.002

ggplot() + theme_bw()+
  geom_histogram(data=TsTv.null,aes(x=Ts),fill="#4286f4")+
  geom_segment(aes(x=Ts.freq,xend=Ts.freq,y=500,yend=90),color="red",arrow=arrow(length=unit(0.5,"cm")),size=2)+
  geom_text(aes(x=Ts.freq,y=600,label=paste("Observed: ",round(Ts.freq,2),"\np-value=",round(p.val,3),sep="")),size=6,fontface="bold",color="red")+
  ggtitle("EGFR Transition Distribution: Uniform Null Model")+
  xlab("Transition Frequency")+ylab("")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```



## Transitions/transversions: null model - conserve transition path frequency

```{r}

n.Ts.path = sum(EGFR.TsTv$class=="Ts")
n.Tv.path = sum(EGFR.TsTv$class=="Tv")

set.seed(5)
nsims = 1e4
TsTv.null = as.data.frame(matrix(nrow=nsims,ncol=2))
colnames(TsTv.null) = c("Ts","Tv")

temp.df = EGFR.TsTv[,c("class","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$class = sample(c("Ts","Tv"),nrow(temp.df),replace=T,prob=c(n.Ts.path,n.Tv.path))
  
  TsTv.null$Ts[i] = sum(temp.df$rel.freq[temp.df$class=="Ts"])
  TsTv.null$Tv[i] = sum(temp.df$rel.freq[temp.df$class!="Ts"])
  
}

Ts.freq = sum(EGFR.TsTv$rel.freq[EGFR.TsTv$class=="Ts"])
p.val = mean(TsTv.null$Ts>=Ts.freq) # 0.52

ggplot() + theme_bw()+
  geom_histogram(data=TsTv.null,aes(x=Ts),fill="#4286f4")+
  geom_segment(aes(x=Ts.freq,xend=Ts.freq,y=500,yend=300),color="red",arrow=arrow(length=unit(0.5,"cm")),size=2)+
  geom_text(aes(x=Ts.freq,y=600,label=paste("Observed: ",round(Ts.freq,2),"\np-value=",round(p.val,3),sep="")),size=6,fontface="bold",color="red")+
  ggtitle("EGFR Transition Distribution: Conserved Path Model")+
  xlab("Transition Frequency")+ylab("")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )


```


## EGFR: Transition/transversion - alternative model

```{r}

n.Ts.alt = sum(LUAD.pyr.sig[names(LUAD.pyr.sig)%in%c("CT","TC")])
n.Tv.alt = sum(LUAD.pyr.sig[!names(LUAD.pyr.sig)%in%c("CT","TC")])


set.seed(5)
nsims = 1e4
TsTv.alt = as.data.frame(matrix(nrow=nsims,ncol=2))
colnames(TsTv.alt) = c("Ts","Tv")

temp.df = EGFR.TsTv[,c("class","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$class = sample(c("Ts","Tv"),nrow(temp.df),replace=T,prob=c(n.Ts.alt,n.Tv.alt))
  
  TsTv.alt$Ts[i] = sum(temp.df$rel.freq[temp.df$class=="Ts"])
  TsTv.alt$Tv[i] = sum(temp.df$rel.freq[temp.df$class!="Ts"])
  
}

Ts.freq = sum(EGFR.TsTv$rel.freq[EGFR.TsTv$class=="Ts"])
p.val = mean(TsTv.alt$Ts>=Ts.freq) # 0.83

ggplot() + theme_bw()+
  geom_histogram(data=TsTv.alt,aes(x=Ts),fill="#4286f4")+
  geom_segment(aes(x=Ts.freq,xend=Ts.freq,y=500,yend=300),color="red",arrow=arrow(length=unit(0.5,"cm")),size=2)+
  geom_text(aes(x=Ts.freq,y=600,label=paste("Observed: ",round(Ts.freq,2),"\np-value=",round(p.val,3),sep="")),size=6,fontface="bold",color="red")+
  ggtitle("EGFR Transition Distribution: Alternative Model")+
  xlab("Transition Frequency")+ylab("")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )


```

## EGFR: Pyrmidine substitutions - null uniform distribution

```{r}

set.seed(5)
nsims = 1e4
pyr.sub.null = as.data.frame(matrix(nrow=nsims,ncol=6))
colnames(pyr.sub.null) = pyr.subs

temp.df = EGFR.mut[,c("pyr.sub","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$pyr.sub = sample(pyr.subs,nrow(temp.df),replace=T,prob=rep(1/6,6))
  
  for (j in 1:length(pyr.subs)) {
    
    pyr.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$pyr.sub==pyr.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

pyr.mat = as.matrix(pyr.sub.null[1:5])
D2 = mahalanobis(pyr.mat,colMeans(pyr.mat),cov(pyr.mat))

pyr.obs = as.vector(pyr.freq.df$rel.freq)
names(pyr.obs) = pyr.freq.df$pyr.sub
D2.obs = mahalanobis(pyr.obs[1:5],colMeans(pyr.mat),cov(pyr.mat))

p.val = mean(D2>=D2.obs) #0.055

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=450,yend=150),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=550,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("EGFR Pyrmidine Substitution: Uniform Null Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(pyr.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(pyr.obs[1:5])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```

## EGFR: Pyrmidine substitutions - conserved path distribution


```{r}

n.path.pyr = table(EGFR.mut$pyr.sub)
n.path.pyr = n.path.pyr[pyr.subs]
n.path.pyr[is.na(n.path.pyr)] = 0
names(n.path.pyr) = pyr.subs

set.seed(5)
nsims = 1e4
pyr.sub.null = as.data.frame(matrix(nrow=nsims,ncol=6))
colnames(pyr.sub.null) = pyr.subs

temp.df = EGFR.mut[,c("pyr.sub","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$pyr.sub = sample(pyr.subs,nrow(temp.df),replace=T,prob=n.path.pyr)
  
  for (j in 1:length(pyr.subs)) {
    
    pyr.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$pyr.sub==pyr.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

pyr.mat = as.matrix(pyr.sub.null)
pyr.mat.f = pyr.mat[,colMeans(pyr.mat)!=0]
pyr.mat.f = pyr.mat.f[,1:(ncol(pyr.mat.f)-1)]
D2 = mahalanobis(pyr.mat.f,colMeans(pyr.mat.f),cov(pyr.mat.f))

pyr.obs = as.vector(pyr.freq.df$rel.freq)
names(pyr.obs) = pyr.freq.df$pyr.sub
pyr.obs.f = pyr.obs[colMeans(pyr.mat)!=0]
pyr.obs.f = pyr.obs.f[1:(length(pyr.obs.f)-1)]
D2.obs = mahalanobis(pyr.obs.f,colMeans(pyr.mat.f),cov(pyr.mat.f))

p.val = mean(D2>=D2.obs) #0.055

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=450,yend=150),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=550,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("EGFR Pyrmidine Substitution: Conserved Path Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(pyr.mat.f,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(pyr.obs.f)))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```

## EGFR: Pyrmidine substitutions - null GC model


```{r}

EGFR.GC.con = 0.54

set.seed(5)
nsims = 1e4
pyr.sub.null = as.data.frame(matrix(nrow=nsims,ncol=6))
colnames(pyr.sub.null) = pyr.subs

temp.df = EGFR.mut[,c("pyr.sub","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$pyr.sub = sample(pyr.subs,nrow(temp.df),replace=T,prob=c(rep(EGFR.GC.con,3),rep(1-EGFR.GC.con,3)))
  
  for (j in 1:length(pyr.subs)) {
    
    pyr.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$pyr.sub==pyr.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

pyr.mat = as.matrix(pyr.sub.null)
pyr.mat.f = pyr.mat[,colMeans(pyr.mat)!=0]
pyr.mat.f = pyr.mat.f[,1:(ncol(pyr.mat.f)-1)]
D2 = mahalanobis(pyr.mat.f,colMeans(pyr.mat.f),cov(pyr.mat.f))

pyr.obs = as.vector(pyr.freq.df$rel.freq)
names(pyr.obs) = pyr.freq.df$pyr.sub
pyr.obs.f = pyr.obs[colMeans(pyr.mat)!=0]
pyr.obs.f = pyr.obs.f[1:(length(pyr.obs.f)-1)]
D2.obs = mahalanobis(pyr.obs.f,colMeans(pyr.mat.f),cov(pyr.mat.f))

p.val = mean(D2>=D2.obs) #0.055
EGFR.pval = p.val

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=450,yend=150),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=550,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("EGFR Pyrmidine Substitution: Null GC Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(pyr.mat.f,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(pyr.obs.f)))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```

## EGFR: Pyrmidine substitutions - control for amino acid substitutions

```{r}

set.seed(5)
nsims = 1e4
pyr.sub.null = as.data.frame(matrix(nrow=nsims,ncol=6))
colnames(pyr.sub.null) = pyr.subs

temp.df = EGFR.mut[,c("pyr.sub","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$pyr.sub = sample(pyr.subs,nrow(temp.df),replace=T,prob=c(68,76,58,64,58,68))
  
  for (j in 1:length(pyr.subs)) {
    
    pyr.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$pyr.sub==pyr.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

pyr.mat = as.matrix(pyr.sub.null[1:5])
D2 = mahalanobis(pyr.mat,colMeans(pyr.mat),cov(pyr.mat))

pyr.obs = as.vector(pyr.freq.df$rel.freq)
names(pyr.obs) = pyr.freq.df$pyr.sub
D2.obs = mahalanobis(pyr.obs[1:5],colMeans(pyr.mat),cov(pyr.mat))

p.val = mean(D2>=D2.obs) #0.055

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=1200,yend=1050),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=1400,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=8,fontface="bold",color="red")+
  ggtitle("Null Model Pyrmidine Substitution:\nEGFR")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=24,face="bold",hjust=0.5),
    axis.title=element_text(size=22,face="bold"),
    axis.text=element_text(size=16)
  )+
  ylim(0,1550)

ggsave("EGFRMahalanobistDist.png",width=6.5,height=6)

pca = prcomp(pyr.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(pyr.obs[1:5])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

write.table(pca.df,"EGFRPCA.csv",row.names=F,col.names=F,sep=",")
write.table(pca.obs,"EGFRPCAobs.csv",row.names=F,col.names=F,sep=",")

```

## EGFR: Pyrmidine substitutions - alternative model

```{r}


set.seed(5)
nsims = 1e4
pyr.sub.alt = as.data.frame(matrix(nrow=nsims,ncol=6))
colnames(pyr.sub.alt) = pyr.subs

temp.df = EGFR.mut[,c("pyr.sub","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$pyr.sub = sample(pyr.subs,nrow(temp.df),replace=T,prob=LUAD.pyr.sig)
  
  for (j in 1:length(pyr.subs)) {
    
    pyr.sub.alt[i,j] = sum(temp.df$rel.freq[temp.df$pyr.sub==pyr.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

pyr.mat = as.matrix(pyr.sub.alt)
pyr.mat.f = pyr.mat[,colMeans(pyr.mat)!=0]
pyr.mat.f = pyr.mat.f[,1:(ncol(pyr.mat.f)-1)]
D2 = mahalanobis(pyr.mat.f,colMeans(pyr.mat.f),cov(pyr.mat.f))

pyr.obs = as.vector(pyr.freq.df$rel.freq)
names(pyr.obs) = pyr.freq.df$pyr.sub
pyr.obs.f = pyr.obs[colMeans(pyr.mat)!=0]
pyr.obs.f = pyr.obs.f[1:(length(pyr.obs.f)-1)]
D2.obs = mahalanobis(pyr.obs.f,colMeans(pyr.mat.f),cov(pyr.mat.f))

p.val = mean(D2>=D2.obs) #0.055

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=450,yend=150),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=550,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("EGFR Pyrmidine Substitution: Alternative Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(pyr.mat.f,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(pyr.obs.f)))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```

## EGFR: Sense strand substitution: uniform null model

```{r}

set.seed(5)
nsims = 1e4
sns.sub.null = as.data.frame(matrix(nrow=nsims,ncol=12))
colnames(sns.sub.null) = sns.subs

temp.df = EGFR.mut[,c("nuc.sub","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$nuc.sub = sample(sns.subs,nrow(temp.df),replace=T,prob=rep(1/12,12))
  
  for (j in 1:length(sns.subs)) {
    
    sns.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$nuc.sub==sns.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

sns.mat = as.matrix(sns.sub.null[1:11])
D2 = mahalanobis(sns.mat,colMeans(sns.mat),cov(sns.mat))

sns.obs = as.vector(sns.freq.df$rel.freq)
names(sns.obs) = sns.freq.df$sns.sub
D2.obs = mahalanobis(sns.obs[1:11],colMeans(sns.mat),cov(sns.mat))

p.val = mean(D2>=D2.obs) # 0.392

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=650,yend=500),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=750,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("EGFR Sense Strand Substitutions: Uniform Null Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(sns.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(sns.obs[1:11])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```





## EGFR: Sense strand substitution: conserve path model

```{r}

n.path.sns = table(EGFR.mut$nuc.sub)
n.path.sns = n.path.sns[sns.subs]
n.path.sns[is.na(n.path.sns)] = 0
names(n.path.sns) = sns.subs

set.seed(5)
nsims = 1e4
sns.sub.null = as.data.frame(matrix(nrow=nsims,ncol=12))
colnames(sns.sub.null) = sns.subs

temp.df = EGFR.mut[,c("nuc.sub","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$nuc.sub = sample(sns.subs,nrow(temp.df),replace=T,prob=n.path.sns)
  
  for (j in 1:length(sns.subs)) {
    
    sns.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$nuc.sub==sns.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

sns.mat = as.matrix(sns.sub.null)
sns.mat.f = sns.mat[,colMeans(sns.mat)!=0]
sns.mat.f = sns.mat.f[,1:(ncol(sns.mat.f)-1)]
D2 = mahalanobis(sns.mat.f,colMeans(sns.mat.f),cov(sns.mat.f))

sns.obs = as.vector(sns.freq.df$rel.freq)
names(sns.obs) = sns.freq.df$sns.sub
sns.obs.f = sns.obs[colMeans(sns.mat)!=0]
sns.obs.f = sns.obs.f[1:(length(sns.obs.f)-1)]
D2.obs = mahalanobis(sns.obs.f,colMeans(sns.mat.f),cov(sns.mat.f))

p.val = mean(D2>=D2.obs) # 0.392

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=650,yend=500),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=750,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("EGFR Sense Strand Substitutions: Conserved Path Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(sns.mat.f,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(sns.obs.f)))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```

## EGFR Trinucleotide pyrmidine substitutions: null model - uniform distribution

```{r}

set.seed(5)
nsims = 5e3
tri.sub.null = as.data.frame(matrix(nrow=nsims,ncol=96))
colnames(tri.sub.null) = tri.subs

temp.df = EGFR.mut[,c("tri.sub","rel.freq")]

# reassign and tally
for (i in 1:nsims) {
  
  temp.df$tri.sub = sample(tri.subs,nrow(temp.df),replace=T,prob=rep(1/96,96))

  for (j in 1:length(tri.subs)) {
    
    tri.sub.null[i,j] = sum(temp.df$rel.freq[temp.df$tri.sub==tri.subs[j]])
    
  }
  
}

# calculate mahalanobis dist

tri.mat = as.matrix(tri.sub.null[1:95])
D2 = mahalanobis(tri.mat,colMeans(tri.mat),cov(tri.mat))

tri.obs = as.vector(tri.freq.df$rel.freq)
names(tri.obs) = tri.freq.df$tri.sub
D2.obs = mahalanobis(tri.obs[1:95],colMeans(tri.mat),cov(tri.mat))

p.val = mean(D2>=D2.obs) # 0.087

ggplot()+theme_bw()+
  geom_histogram(aes(D2),bins=50,fill="#4286f4")+
  geom_segment(aes(x=D2.obs,xend=D2.obs,y=500,yend=50),color="red",arrow = arrow(length = unit(0.5,"cm")),size=2)+
  geom_text(aes(x=D2.obs,y=600,label=paste("Observed:",round(D2.obs,2),"\np-value =",round(p.val,3))),size=6,fontface="bold",color="red")+
  ggtitle("EGFR Trinucleotide Pyrmidine Substitutions: Uniform Null Model")+
  xlab("Mahalanobis Distance")+ylab("Frequency")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

pca = prcomp(tri.mat,scale=T)
pca.obs = predict(pca,newdata = t(data.frame(tri.obs[1:95])))

pca.df = as.data.frame(pca$x)
ggplot(pca.df)+theme_bw()+
  geom_point(aes(PC1,PC2),color="gray50",size=3,alpha=0.4)+
  geom_point(aes(pca.obs[1],pca.obs[2]),color="red",size=4)+
  xlab("Principle Component 1")+ylab("Principle Component 2")+
  ggtitle("Principle Component Analysis")+
  theme(
    plot.title=element_text(size=20,face="bold",hjust=0.5),
    axis.title=element_text(size=18,face="bold"),
    axis.text=element_text(size=16)
  )

```

